---
title: | 
   **Defining protein markers for *Candida albicans* extracellular vesicles**
author: | 
   *Charlotte Dawson^a^, Mark Bleackley^a^, Donovan Garcia Ceron^a^, 
   Marilyn Anderson^a^*
date: | 
   *^a^Department of Biochemistry and Genetics, La Trobe Institute for 
   Molecular Science, La Trobe University, Victoria, 3086, Australia*
output: 
    bookdown::pdf_document2:
      latex_engine: xelatex
      fig_caption: yes
      number_sections: yes
      toc: no
      keep_tex: no
    bookdown::word_document2:
      pandoc_args:
        '--lua-filter=components/pagebreak.lua'
      reference_docx: "components/draft_template.docx"
      fig_caption: yes
bibliography: "components/references.bib" 
csl: "components/molecular-and-cellular-proteomics.csl"
link-citations: yes
linkcolor: black
fontsize: 12pt
geometry: margin=1in
classoption: letterpaper
mainfont: Times New Roman
linestretch: 1.5
header-includes:
  - \usepackage{titling}
  - \pretitle{\begin{center}\fontsize{14bp}{14bp}\selectfont}
  - \posttitle{\par\end{center}}
  - \preauthor{\begin{center}\fontsize{12bp}{12bp}\selectfont}
  - \postauthor{\par\end{center}}
  - \predate{\begin{center}\fontsize{10bp}{10bp}\selectfont}
  - \postdate{\par\end{center}}
  - \usepackage{fancyhdr}
  - \usepackage{lipsum}
  - \pagestyle{fancy}
  - \fancyhead[L]{This is the running title it is rather long}
  - \setlength{\headheight}{15pt}
  - \usepackage{float}
  - \let\origfigure\figure
  - \let\endorigfigure\endfigure
  - \renewenvironment{figure}[1][2]{\expandafter\origfigure\expandafter[H]}
    {\endorigfigure}
  - \usepackage{caption}
  - \captionsetup{font={normalsize,stretch=1.3},labelfont=bf}
  - \usepackage{threeparttable}
---

**Running title:** Running title goes here

**Corresponding author:**

\newpage

```{r setup, include=FALSE}
# set knit options
knitr::opts_knit$set(progress = TRUE, 
                     verbose = TRUE)

# set chunk options
knitr::opts_chunk$set(collapse = TRUE,
                      warning = FALSE,
                      message = FALSE,
                      echo = FALSE,
                      comment = "#>")

# load packages
library(candidaev)
library(tidyNano)
library(limma)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(kableExtra)

# set seed for random number generation
set.seed(1)

# reference external R script with protein function lists
knitr::read_chunk('components/protein_funs.R')

# define colour palette to distinguish between strains
# colours = yellow, blue, green, pink
# levels = yeast, ATCC90028, ATCC10231, biofilm
st_pal <- c("#E69F00", "#56B4E9", "#009E73", "#CC79A7")

# define colour palette to distinguish between EVs and whole cell lysate (WCL)
# colours = grey, red, blue
# levels = not significant, EV, WCL
ev_pal <- c("#999999", "#D55E00", "#0072B2")
```

```{r yeast, include=FALSE}
# filter out reverse, contaminant proteins, proteins with <2 unique peptides
# extract LFQ intensity columns and UniProt accessions 
# log2 transform LFQ intensities
yeast_lfq <- yeast %>% 
  filter(Reverse != "+", 
         Potential.contaminant != "+", 
         Unique.peptides >= 2) %>% 
  convert_lfq(., yeast_exp)

# drop EV_1 and WCL_1 samples
yeast_lfq2 <- yeast_lfq[, !colnames(yeast_lfq) %in% c("EV_1", "WCL_1")]

# filter for proteins quantified in min 2/3 reps of EV or WCL
yeast_filt2 <- filter_na2(yeast_lfq2, logic = "or", op = "<=", 
                          pat1 = "EV", val1 = 1, 
                          pat2 = "W", val2 = 1)

# normalise LFQ intensities
yeast_norm2 <- normalizeCyclicLoess(yeast_filt2)

# filter for proteins identified in min 1/3 reps of EV
# define 'EV proteins'
y_ev <- filter_na(yeast_norm2, op = "<=", 
                  pat = "EV", val = 2)

# filter for proteins identified in min 1/3 reps of WCL
# define 'WCL proteins'
y_wcl <- filter_na(yeast_norm2, op = "<=", 
                   pat = "W", val = 2)

# can't impute with so many missing values
# separate EV/WCL exclusive proteins (don't impute) from 
# EV/WCL common proteins (impute)
yeast_excl <- filter_na2(yeast_norm2, logic = "or", op = "==", 
                         pat1 = "EV", val1 = 3, 
                         pat2 = "W", val2 = 3)

yeast_both <- filter_na2(yeast_norm2, logic = "and", op = "<=", 
                         pat1 = "EV", val1 = 2, 
                         pat2 = "W", val2 = 2)

# proteins with missing values tend to have lower intensity
# therefore proteins are MNAR, close to detection limit
# use left censored imputation method
yeast_imp <- impute_QRILC(yeast_both)

# recombine imputed proteins and non-imputed proteins in matrix
yeast_de <- rbind(yeast_excl, yeast_imp)

# see limma user guide section 9.2 for more info about DE
# create design matrix
y_samp <- data.frame(T = rep(c("EV", "WCL"), each = 3))

y_design <- stats::model.matrix(~ 0 + T, data = y_samp)
colnames(y_design) <- c("EV", "WCL")

# define sample comparisons of interest
y_contrasts <- c("EV - WCL")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
y_efit <- limma_eBayes(yeast_de, design = y_design, contrasts = y_contrasts)

# extract DE results
yeast_res <- get_results(efit = y_efit, 
                         mat = yeast_de, 
                         p_val = 0.01, 
                         lfc = 0, 
                         type = "individual")[[1]]
```

```{r biofilm, include=FALSE}
# filter out reverse, contaminant proteins, proteins with <2 unique peptides
# extract LFQ intensity columns and UniProt accessions 
# log2 transform LFQ intensities
biofilm_lfq <- biofilm %>% 
  filter(Reverse != "+", 
         Potential.contaminant != "+", 
         Unique.peptides >= 2) %>% 
  convert_lfq(., biofilm_exp)

# filter for proteins quantified in min 4/5 reps of EV or WCL
biofilm_filt <- filter_na2(biofilm_lfq, logic = "or", op = "<=", 
                           pat1 = "EV", val1 = 1, 
                           pat2 = "W", val2 = 1)

# normalise LFQ intensities
biofilm_norm <- normalizeCyclicLoess(biofilm_filt)

# filter for proteins identified in min 1/5 reps of EV
# define 'EV proteins'
b_ev <- filter_na(biofilm_norm, op = "<=", 
                  pat = "EV", val = 4)

# filter for proteins identified in min 1/5 reps of WCL
# define 'WCL proteins'
b_wcl <- filter_na(biofilm_norm, op = "<=", 
                   pat = "W", val = 4)

# can't impute with so many missing values
# filter for proteins with 4-5 NA values in EV or WCL (don't impute)
biofilm_excl <- filter_na2(biofilm_norm, logic = "or", op = ">=", 
                         pat1 = "EV", val1 = 4, 
                         pat2 = "W", val2 = 4)

# filter for proteins with min 3 valid values in EV and WCL (impute)
biofilm_both <- filter_na2(biofilm_norm, logic = "and", op = "<=", 
                         pat1 = "EV", val1 = 3, 
                         pat2 = "W", val2 = 3)

# proteins with missing values tend to have lower intensity
# therefore proteins are MNAR, close to detection limit
# use left censored imputation method
biofilm_imp <- impute_QRILC(biofilm_both)

# recombine imputed proteins and non-imputed proteins in matrix
biofilm_de <- rbind(biofilm_excl, biofilm_imp)

# see limma user guide section 9.2 for more info about DE
# create design matrix
b_samp <- data.frame(T = rep(c("EV", "WCL"), each = 5))

b_design <- stats::model.matrix(~ 0 + T, data = b_samp)
colnames(b_design) <- c("EV", "WCL")

# define sample comparisons of interest
b_contrasts <- c("EV - WCL")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
b_efit <- limma_eBayes(biofilm_de, design = b_design, contrasts = b_contrasts)

# extract DE results
biofilm_res <- get_results(efit = b_efit, 
                           mat = biofilm_de, 
                           p_val = 0.01, 
                           lfc = 0, 
                           type = "individual")[[1]]
```

```{r atcc, include=FALSE}
# filter out reverse, contaminant proteins, proteins with <2 unique peptides
# extract LFQ intensity columns and UniProt accessions 
# log2 transform LFQ intensities
atcc_lfq <- atcc %>% 
  filter(Reverse != "+", 
         Potential.contaminant != "+", 
         Unique.peptides >= 2) %>% 
  convert_lfq(., atcc_exp)

# filter for proteins quantified in min 2/3 reps of: 
# A1 EV or A1 WCL or A9 EV or A9 WCL
atcc_filt <- filter_na4(atcc_lfq, "or", "<=", 
                        pat1 = "A10231_EV", 1, 
                        pat2 = "A10231_W", 1, 
                        pat3 = "A90028_EV", 1, 
                        pat4 = "A90028_W", 1)

# normalise LFQ intensities
atcc_norm <- normalizeCyclicLoess(atcc_filt)

# filter for proteins identified in min 1/3 reps of 10231 EV
# define 'ATCC10231 EV proteins'
atcc1_ev <- filter_na(atcc_norm, op = "<=", 
                      pat = "A10231_EV", val = 2)

# filter for proteins identified in min 1/3 reps of 10231 WCL
# define 'ATCC10231 WCL proteins'
atcc1_w <- filter_na(atcc_norm, op = "<=", 
                     pat = "A10231_W", val = 2)

# filter for proteins identified in min 1/3 reps of 90028 EV
# define 'ATCC90028 EV proteins'
atcc9_ev <- filter_na(atcc_norm, op = "<=", 
                      pat = "A90028_EV", val = 2)

# filter for proteins identified in min 1/3 reps of 90028 WCL
# define 'ATCC90028 WCL proteins'
atcc9_w <- filter_na(atcc_norm, op = "<=", 
                      pat = "A90028_W", val = 2)

# can't impute with so many missing values
# filter for exclusive to 10231 EV or WCL or 90028 EV or WCL (don't impute)
atcc_excl <- filter_na4(atcc_norm, logic = "or", op = "==", 
                        pat1 = "A10231_EV", val1 = 3, 
                        pat2 = "A10231_W", val2 = 3, 
                        pat3 = "A90028_EV", val3 = 3, 
                        pat4 = "A90028_W", val4 = 3)

# filter for proteins with min 1 valid value in all 4 sample types (impute)
atcc_both <- filter_na4(atcc_norm, logic = "and", op = "<=", 
                        pat1 = "A10231_EV", val1 = 2, 
                        pat2 = "A10231_W", val2 = 2, 
                        pat3 = "A90028_EV", val3 = 2, 
                        pat4 = "A90028_W", val4 = 2)

# proteins with missing values tend to have lower intensity
# therefore proteins are MNAR, close to detection limit
# use left censored imputation method
atcc_imp <- impute_QRILC(atcc_both)

# recombine imputed proteins and non-imputed proteins in matrix
atcc_de <- rbind(atcc_excl, atcc_imp)
colnames(atcc_de) <- stringi::stri_replace_all_regex(colnames(atcc_de), 
                                                     c("A10231_", "A90028_"), 
                                                     c("A1_", "A9_"), 
                                                     vectorize_all = FALSE)

# see limma user guide section 9.2 for more info about DE
# create design matrix
atcc_samp <- data.frame(T = rep(c("A1_EV", "A1_WCL", 
                                  "A9_EV", "A9_WCL"), each = 3))

atcc_design <- stats::model.matrix(~ 0 + T, data = atcc_samp)
colnames(atcc_design) <- c("A1_EV", "A1_W", "A9_EV", "A9_W")

# define sample comparisons of interest
atcc_contrasts <- c("A1_EV - A1_W", "A9_EV - A9_W")

# make all pair-wise comparisons for specified contrasts
# and perform limma::eBayes()
atcc_efit <- limma_eBayes(atcc_de, atcc_design, atcc_contrasts)

# extract overall results
atcc_overall <- get_results(efit = atcc_efit, mat = atcc_de, 
                            p_val = 0.01, lfc = 0, type = "overall")

# extract results for ATCC strains
a1_res <- get_results(efit = atcc_efit, 
                      mat = atcc_de, 
                      p_val = 0.01, 
                      lfc = 0, 
                      type = "individual")[[1]]

a9_res <- get_results(efit = atcc_efit, 
                      mat = atcc_de, 
                      p_val = 0.01, 
                      lfc = 0, 
                      type = "individual")[[2]]
```


# Abbreviations

EV - extracellular vesicle

\newpage

# Abstract

The abstract goes here. Here is a citation [@liao2012].

Here is another citation [@gil-bona2015].

And another [@lin2014].

\newpage

# Introduction

Here is a citation [@marwick2017].

# Experimental procedures

## Subsection 1

## Subsection 2

# Results

# Discussion

# Acknowledgements

# Data availability

# Acknowledgements

\newpage
# References 
<div id="refs"></div>

\newpage

# Tables

```{r table1Data, include=FALSE}
# make list of EV sig. enriched and exclusive proteins for each strain
marker_comp <- list("DAY Y" = yeast_res %>% 
                      filter(group == "EV up" | group == "EV ex") %>% 
                      pull(CGD_gene_name), 
                    "DAY B" = biofilm_res %>% 
                      filter(group == "EV up" | group == "EV ex") %>% 
                      pull(CGD_gene_name), 
                    "ATCC90028" = a9_res %>% 
                      filter(group == "A9_EV up" | group == "A9_EV ex") %>% 
                      pull(CGD_gene_name), 
                    "ATCC10231" = a1_res %>% 
                      filter(group == "A1_EV up" | group == "A1_EV ex") %>% 
                      pull(CGD_gene_name))

# get venn segments
marker_comp_tab <- plot_venn(vlist = marker_comp, 
                             use_uniprot = FALSE, 
                             type = "df")

# extract protein names from venn overlap
marker_candidates <- marker_comp_tab %>% 
  filter(apply(.[, 1:4], 1, sum) == 4) %>% 
  pull(..values..) %>% 
  unlist() %>% 
  as.data.frame(stringsAsFactors = FALSE) %>% 
  rename(Name = 1)

# construct table 1
marker_longlist <- marker_candidates %>% 
  mutate(Function = match_id(Name, uniprot, "CGD_gene_name", "Protein_name"),
         DAY.Y = match_id(Name, yeast_res, "CGD_gene_name", "logFC", 
                          str_split = FALSE), 
         A9 = match_id(Name, a9_res, "CGD_gene_name", "logFC", 
                       str_split = FALSE), 
         A1 = match_id(Name, a1_res, "CGD_gene_name", "logFC", 
                       str_split = FALSE), 
         DAY.B = match_id(Name, biofilm_res, "CGD_gene_name", "logFC", 
                          str_split = FALSE), 
         TM = match_id(Name, uniprot, "CGD_gene_name", "Transmembrane"), 
         SP = match_id(Name, uniprot, "CGD_gene_name", "Signal_peptide")) %>% 
  mutate_at(.vars = vars(DAY.Y, DAY.B, A1, A9), 
            .funs = list(~ sprintf(as.numeric(.), fmt = "%#.2f"))) %>% 
  mutate_at(.vars = vars(DAY.Y, DAY.B, A1, A9), 
            .funs = list(~ case_when(. == "NA" ~ "ex", 
                                     TRUE ~ as.character(.)))) %>% 
  mutate(TM = stringr::str_count(TM, "Helical"), 
         SP = case_when(grepl("SIGNAL", SP) ~ "Y", 
                        TRUE ~ ""), 
         TM = case_when(TM == 0 ~ "", 
                        TRUE ~ as.character(TM))) %>% 
  rename("DAY Y" = DAY.Y, "DAY B" = DAY.B) %>% 
  arrange(Name)

# load vector of protein functions from external R script
<<fun_list1>>

# add proper protein functions to table 1
# also add vesicle depleted supernatant info and 
# group proteins according to localisation
marker_longlist <- marker_longlist %>% 
  mutate(Function = function_desc1, 
         VDM = c("Y", rep("",3), rep("Y", 2), "", "Y", rep("", 8), rep("Y", 2), 
                 rep("", 4), rep("Y", 3), rep("", 2), rep("Y", 2), rep("", 5), 
                 "Y", rep("", 7), "Y"), 
         pack_row = c(5, rep(1,3), rep(5,2), 1, 2, rep(1,3), rep(1,3), 4, 2, 
                      rep(5,2), 1, rep(2,2), 4, 1, rep(5,2), 1, rep(5,3), 1, 
                      4, rep(1,3), 5, 1, 1, 3, 1, 3, 2, 3, 5)) %>% 
  arrange(pack_row)

# fix number of TM domains for CDR1;CDR2
marker_longlist[2, 7] <- "12;12" 

# UniProt says SUR7 has 3 TM domains and a signal peptide
# much of literature says SUR7 has 4 TM domains
# TOPCONS2 predicts SUR7 has 4 TM domains and no signal peptide
# fix SUR7 entries
marker_longlist[19, 7] <- "4"
marker_longlist[19, 8] <- ""
```

<!-- assign markdown references for table caption -->
(ref:table1) [@gil-bona2015]
(ref:footnote1) [@skrzypek2017]
(ref:footnote2) [@almagroarmenteros2017]
(ref:footnote3) [@tsirigos2015]

```{r table1}
tab1_cap <- "\\textbf{Candidate positive protein markers for 
             \\textit{C. albicans} EVs.} This list of proteins consists of 
             those that were found to be exclusive to EVs or significantly 
             enriched in EVs across the four \\textit{C. albicans} strains 
             examined in this study. Proteins are grouped according to their 
             subcellular localisation as annotated in the Candida Genome 
             Database (candidagenome.org) (ref:footnote1). 
             The log\\textsubscript{2} ratio of the abundance (mean MaxQuant 
             LFQ intensity) of each protein in EVs compared to whole cell 
             lysate (WCL) for each strain is listed. \`\`ex\" indicates where 
             a protein was only quantified in the EV fraction and not the WCL 
             for that strain. The \`\`TM\" column indicates the number of 
             transmembrane domains for each protein as annotated in UniProtKB. 
             \`\`SP\" indicates whether a protein is annotated as having a 
             signal peptide according to UniProtKB. \`\`VDM\" shows whether a 
             protein has previously been detected in vesicle-depleted culture 
             media (i.e. the proteins may also be in the soluble secretome) 
             (ref:table1). Underlined proteins are those identified as the 
             best candidates for positive EV markers according to the 
             criteria depicted in Figure \\ref{fig:figure1}."

# draw table 1
knitr::kable(marker_longlist[ ,-10],
             caption = tab1_cap, 
             format = "latex", 
             booktabs = TRUE, 
             longtable = TRUE, 
             linesep = "", 
             escape = FALSE)  %>% 
  row_spec(row = c(1, 5, 6, 11, 12, 15:20, 22:27, 39), underline = TRUE) %>% 
  add_header_above(c(" " = 2, 
                     "log\\\\textsubscript{2}(fold change) EV vs WCL" = 4, 
                     " " = 2), escape = FALSE) %>% 
  kable_styling(font_size = 8) %>% 
  pack_rows("Plasma membrane", 1, 20) %>% 
  pack_rows("Golgi, ER, secretory pathway", 21, 25) %>% 
  pack_rows("Vacuole", 26, 28) %>% 
  pack_rows("Mitochondria", 29, 31) %>% 
  pack_rows("Cell wall, cell surface", 32, 43) %>% 
  footnote(alphabet = c("Protein localisation was inferred from sequence 
                         similarity with \\\\textit{S. cerevisiae} ortholog as 
                         annotated in the Candida Genome Database 
                        (ref:footnote1)", 
                        "Protein localisation was predicted using DeepLoc-1.0 
                        (ref:footnote2)", 
                        "Presence of transmembrane domains and absense of a 
                        signal peptide was predicted using TOPCONS2 
                        (ref:footnote3)"), 
           escape = FALSE, 
           threeparttable = TRUE)
```

\newpage

# Figures

```{r figure1, fig.cap=f1_cap, fig.width=5.45, fig.height=6}
# construct figure 1
f1_cap <- "\\textbf{Schematic of the workflow used for the isolation and 
           proteomic analysis of \\textit{C. albicans} EVs and the selection 
           of EV protein markers.} Steps 1 to 6 were performed for each of the 
           four strains examined in this study: DAY286 yeast, ATCC90028 yeast, 
           ATCC10231 yeast, and DAY286 biofilm. The results from the four 
           seperate analyses were combined (step 7) to identify proteins which 
           commonly appeared as EV marker candidates. The MISEV2018 criteria 
           for protein content-based EV characterisation and other 
           considerations were used to select the best \\textit{C. albicans} 
           EV marker proteins."

f1 <- png::readPNG("figures/figure1.png")

# draw figure 1
ggplot() + 
  background_image(f1)
```

\newpage

```{r figure2Data, include=FALSE}
# construct data.frame with nta experiment info
nta_metadata <- data.frame(id = c("ATCC10231_1", "ATCC10231_2", "ATCC10231_3", 
                                  "ATCC90028_1", "ATCC90028_2", "ATCC90028_3", 
                                  "Biofilm_1", "Biofilm_2", "Biofilm_3", 
                                  "Biofilm_4", "Biofilm_5", 
                                  "DAY286_1", "DAY286_2", "DAY286_3"), 
                           protein_conc = c(0.192, 0.282, 0.287, 
                                            0.424, 0.716, 0.584, 
                                            0.204, 0.198, 0.297, 0.193, 0.416, 
                                            0.437, 0.360, 0.447), 
                           stringsAsFactors = FALSE)

# average the nta technical replicate data
nta_techavg <- nta %>% 
  nanolyze(particle_size, Strain, Bio_rep, Dilution, 
           name = "Tech_rep", 
           param_var = True_count)

# average the nta biological replicate data
nta_bioavg <- nta_techavg %>% 
  nanolyze(particle_size, Strain,
           name = "Bio_rep", 
           param_var = Tech_rep_mean) %>% 
  mutate(facet = case_when(Strain == "Biofilm" ~ "second", TRUE ~ "first"))

# determine the total particle concentration for each biological replicate
nta_total <- nta_techavg %>% 
  nanocount(Strain, Bio_rep, Dilution, 
            param_var = Tech_rep_mean) %>% 
  tidyr::unite("id", Strain, Bio_rep) %>% 
  mutate(protein_conc = as.numeric(match_id(id, nta_metadata, 
                                            "id", "protein_conc")), 
         od = match_id(id, nta_metadata, "id", "OD600"), 
         part_per_prot = Particle_count / protein_conc / 1000) %>% 
  tidyr::separate(col = id, into = c("Strain", "Bio_rep"), sep = "_", 
                  remove = TRUE)

# use ANOVA with Tukey HSD to compare mean particle per ug protein 
# ratio across the strains
nta_total_tukey <- aov(part_per_prot ~ Strain, data = nta_total) %>% 
  tukey_hsd()
```


```{r figure2plots, include=FALSE}
# construct figure 2a
f2a_img <- png::readPNG("figures/figure2a.png")

f2a <- ggplot() + 
  background_image(f2a_img) + 
  theme(plot.margin = margin(t = 0.3, l = 0, r = 0, b = 0, unit = "inches"))

# construct figure 2b
f2b <- nta_bioavg %>% 
  ggplot(aes(x = particle_size, y = Bio_rep_mean, colour = Strain)) + 
  scale_colour_manual(values = st_pal[c(3, 2, 4, 1)], 
                      breaks = c("DAY286", "ATCC90028", 
                                 "ATCC10231", "Biofilm"), 
                      labels = c("DAY286 Y", "ATCC90028", 
                                 "ATCC10231", "DAY286 B")) + 
  geom_vline(xintercept = 100, colour = "#999999", linetype = "dashed") +
  geom_line(size = 0.5) + 
  facet_wrap(~ facet, scales = "free", ncol = 2, nrow = 1) + 
  theme_classic() + 
  theme(strip.background = element_blank(), 
        strip.text = element_blank(), 
        legend.position = c(0.88, 0.6), 
        legend.title = element_blank(), 
        plot.margin = margin(t = 0.2, l = 0, r = 0.2, b = 0, unit = "inches"), 
        axis.text.x = element_text(colour = "black")) + 
  xlab(label = "Particle size (nm)") + 
  ylab(label = "Particles / mL")

# construct figure 2c
nta_total$Strain <- factor(nta_total$Strain, 
                           levels = c("DAY286", "ATCC90028", 
                                      "ATCC10231", "Biofilm"))

f2c <- nta_total %>% 
  ggboxplot(x = "Strain", y = "part_per_prot", 
            color = "Strain", palette = st_pal, add = "jitter", 
            add.params = list(color = "black"), 
            shape = 15,
            ylab = expression("Particles /"~paste(mu,g)~"protein"))
f2c <- ggpar(f2c, 
             yscale = "log10", 
             format.scale = TRUE, 
             ylim = c(8e8, 8e10),
             font.y = 10, 
             legend = "none") + 
  stat_pvalue_manual(nta_total_tukey[c(2,4,6), ], label = "p.adj", 
                     y.position = c(10.45, 10.65, 10.85), 
                     tip.length = 1e-2, 
                     label.size = 3) + 
  theme(axis.title.x = element_blank(), 
        plot.margin = margin(t = 0.2, l = 0.2, r = 0, b = 0.15, 
                             unit = "inches")) + 
  scale_x_discrete(labels = c("ATCC90028" = "A9", 
                              "ATCC10231" = "A1", 
                              "Biofilm" = "B", 
                              "DAY286" = "Y"))
```


```{r figure2, fig.cap=f2_cap, fig.width=7.1, fig.height=4}
# draw figure 2
ggarrange(f2a, 
          ggarrange(f2b, f2c, 
                    labels = c("B", "C"), 
                    ncol = 2,
                    nrow = 1, 
                    widths = c(1.8,1)), 
          labels = c("A", ""), 
          nrow = 2, 
          ncol = 1, 
          heights = c(1,1))

f2_cap <- "\\textbf{Characterisation and quantification of EVs from different 
           \\textit{C. albicans} strains.} \\textit{(A)} Representative TEM 
           images of EVs isolated from 4 different \\textit{C. albicans} 
           strains. Scale bar indicates 0.5 $\\mu$m. \\textit{(B)} Size 
           distribution of DAY286 yeast (n = 3), ATCC90028 yeast (n = 3), 
           ATCC10231 yeast (n = 3), and DAY286 biofilm (n = 5) EVs as measured 
           by nanoparticle tracking analysis. Line plots shown are the average 
           of the biological replicates and the dashed grey line indicates 100 
           nm. Plots for the individual biological replicates are presented in 
           supplemental Figure S1. \\textit{(C)} Comparison of the ratio of 
           particle concentration to protein concentration of the EV fraction, 
           (particles/mL) $\\div$ ($\\mu$g protein/mL), across the four EV 
           sources. Each dot represents one biological replicate. Sample means 
           were compared using ANOVA followed by Tukey's HSD \\textit{post-hoc} 
           test. P-values indicating significant differences are shown. The 
           particle and protein concentrations for each biological replicate 
           are provided in supplemental Table S1."
```

\newpage

```{r figure3Yeast, include=FALSE}
# compare DAY286 yeast EV WCL proteins
f3a1 <- plot_venn(list("EV" = y_ev, "WCL" = y_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8,
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  margin = 0.05)

# DAY286 yeast venn
f3a1 <- as_ggplot(f3a1) + 
  theme(plot.margin = unit(c(1,0.5,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, 
           label = "bold('DAY286 yeast')", parse = TRUE)

# generate volcano plot
f3b1 <- plot_volcano(data = yeast_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched", 
                                 "WCL enriched", 
                                 "Not significant"), 
                      values = unname(ev_pal[c(2,1,3)]), 
                      breaks = c("EV up", "WCL up", "not sig"), 
                      name = element_blank()) + 
  annotate("text", x = 8.5, y = 0.5, label = "bold('DAY286 Y')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('88')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('88')", parse = TRUE, 
           colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('318')", parse = TRUE, 
           colour = unname(ev_pal[1]))
```



```{r figure3A9, include=FALSE}
# compare ATCC90028 EV WCL proteins
f3a2 <- plot_venn(list("EV" = atcc9_ev, "WCL" = atcc9_w), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans",
                  cex = 0.8,
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.55)), 
                  ext.percent = c(30,10,0),
                  ext.pos = c(45,315),
                  ext.dist = -0.05,
                  ext.length = 0.5,
                  margin = 0.05)

# ATCC90028 venn
f3a2 <- as_ggplot(f3a2) + 
  theme(plot.margin = unit(c(1,0,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('ATCC90028')", parse = TRUE)

# generate volcano plot
f3b2 <- plot_volcano(data = a9_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched (99)", "WCL enriched (130)", 
                                 "Not significant (744)"),
                      values = unname(ev_pal[c(2,3,1)])) + 
  annotate("text", x = 8, y = 0.5, label = "bold('ATCC90028')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('99')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('130')", 
           parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('744')", 
           parse = TRUE, colour = unname(ev_pal[1]))
```



```{r figure3A1, include=FALSE}
# compare ATCC10231 EV WCL proteins
f3a3 <- plot_venn(list("EV" = atcc1_ev, "WCL" = atcc1_w), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8,
                  cat.fontfamily = "sans",
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  ext.percent = c(0,20,0),
                  ext.pos = 45,
                  ext.dist = -0.05,
                  ext.length = 0.5,
                  margin = 0.05)

# ATCC10231 venn
f3a3 <- as_ggplot(f3a3) + 
  theme(plot.margin = unit(c(1,0.5,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('ATCC10231')", parse = TRUE)
  
# generate volcano plot
f3b3 <- plot_volcano(data = a1_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched (108)", "WCL enriched (142)", 
                                 "Not significant (435)"),
                      values = unname(ev_pal[c(2,3,1)])) + 
  annotate("text", x = 8, y = 0.5, label = "bold('ATCC10231')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('106')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('134')", 
           parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('445')", 
           parse = TRUE, colour = unname(ev_pal[1]))
```



```{r figure3Biofilm, include=FALSE}
# compare DAY286 biofilm EV WCL proteins
f3a4 <- plot_venn(list("EV" = b_ev, "WCL" = b_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8, 
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  ext.text = TRUE,
                  ext.percent = c(20,10,0),
                  ext.pos = c(45,315),
                  ext.dist = -0.1,
                  ext.length = 0.5,
                  margin = 0.05)
# DAY286 biofilm venn
f3a4 <- as_ggplot(f3a4) + 
  theme(plot.margin = unit(c(1,0,1,0), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('DAY286 biofilm')", 
           parse = TRUE)

# generate volcano plot
f3b4 <- plot_volcano(data = biofilm_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched (95)", "WCL enriched (79)", 
                                 "Not significant (484)"), 
                      values = unname(ev_pal[c(2,1,3)]), 
                      breaks = c("EV up", "WCL up", "not sig")) + 
  annotate("text", x = 8.5, y = 0.5, label = "bold('DAY286 B')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('95')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('79')", 
           parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('484')", 
           parse = TRUE, colour = unname(ev_pal[1]))
```



```{r figure3, fig.cap=f3_cap, fig.width=7.1, fig.height=6}
# draw figure 3
ggarrange(ggarrange(f3a1, f3a2, f3a3, f3a4, 
                    labels = c("A", "", "", ""), 
                    ncol = 4, 
                    nrow = 1, 
                    align = "v"), 
          ggarrange(f3b1, f3b2,
                    f3b3, f3b4, 
                    labels = c("B", rep("", 3)), 
                    ncol = 2, 
                    nrow = 2, 
                    align = "hv", 
                    common.legend = TRUE, 
                    legend = "right", 
                    widths = c(1,1), 
                    vjust = 1), 
          ncol = 1,
          nrow = 2, 
          heights = c(1,2))

f3_cap <- "\\textbf{Differential abundance analysis of proteins identified in 
           \\textit{C. albicans} EVs and whole cell lysates (WCL).} 
           \\textit{(A)} Venn diagrams comparing the EV and WCL proteomes from 
           four \\textit{C. albicans} strains; DAY286 yeast (n = 3), ATCC90028 
           yeast (n = 3), ATCC10231 yeast (n = 3), and DAY286 biofilm (n = 5). 
           Whole cell lysates were prepared from the EV source cells. Proteins 
           were quantified (MaxQuant LFQ intensity) in minimum 2/3 (yeast) and 
           4/5 (biofilm) biological replicates of EV or WCL. \\textit{(B)} 
           Volcano plots depicting significantly enriched EV or WCL proteins. 
           Differential abundance analysis was performed by comparing the mean 
           normalised LFQ intensities of proteins identified in both EV and 
           WCL (i.e. proteins in the venn overlap) using the Bioconductor 
           package \\textit{limma} [@cox2014; @ritchie2015]. Significantly 
           enriched proteins were identified using a Benjamini-Hochberg 
           adjusted p-value cut off of 0.01. Counts of significant and 
           non-significant proteins are indicated on each graph. Proteins with 
           a log\\textsubscript{2}(FC) greater than 8 are labelled. Data tables 
           underlying the venn diagrams and volcano plots are provided in 
           supplemental File S2."
```

\newpage

```{r figure4Data, include=FALSE}
# get list of DAY286 yeast EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f4_y_go <- marker_comp[1] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  gsub("CHS3", "CaO19.4937", .) %>% 
  gsub("PEP1", "CaO19.3767", .)

# get list of DAY286 biofilm EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f4_b_go <- marker_comp[2] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017"), 
                                  vectorize_all = FALSE)

# get list of ATCC90028 EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f4_a9_go <- marker_comp[3] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5", "^MNN2$", 
                                    "MNN22", "MNN24", "ECM331", 
                                    "CAM1-1", "FCY21"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017", "CaO19.2347", 
                                    "CaO19.3803", "CaO19.1995", "CaO19.4255", 
                                    "CaO19.2651", "CaO19.8937"), 
                                  vectorize_all = FALSE)
f4_a9_go <- f4_a9_go[!f4_a9_go %in% c("orf19.7397.1", "orf19.7398")]

# get list of ATCC10231 EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f4_a1_go <- marker_comp[4] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5", "^MNN2$", 
                                    "MNN22", "MNN24", "ECM331", 
                                    "CAM1-1"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017", "CaO19.2347", 
                                    "CaO19.3803", "CaO19.1995", "CaO19.4255", 
                                    "CaO19.2651"), 
                                  vectorize_all = FALSE)
```

```{r figure4Plots, include=FALSE}
# import FungiFun2 GO enrichment results
f4_y_go_table <- process_fungifun("figures/f4_GO_DAY_Y.csv")
f4_b_go_table <- process_fungifun("figures/f4_GO_DAY_B.csv")
f4_a1_go_table <- process_fungifun("figures/f4_GO_ATCC10231.csv")
f4_a9_go_table <- process_fungifun("figures/f4_GO_ATCC90028.csv")

# define the colour palette
f4_pal <- c("#6D2077", "#E4002B", "#FFB81C")

# GO term bar plots
f4_y <- plot_go_bar(f4_y_go_table, c(1:4, 49,52:53,55, 77,79:81), f4_pal) + 
  annotate("text", x = 12, y = 65, label = "bold('DAY286 Y')", parse = TRUE) + 
  labs(fill = "GO domain") + 
  theme(legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10))

f4_b <- plot_go_bar(f4_b_go_table, c(1:4, 69,71:73, 96:97, 100, 102), f4_pal) + 
  annotate("text", x = 12, y = 60, label = "bold('DAY286 B')", parse = TRUE)

f4_a9 <- plot_go_bar(f4_a9_go_table, c(1:4, 35,37:38,42, 55:56,58:59), f4_pal) + 
  annotate("text", x = 12, y = 70, label = "bold('ATCC90028')", parse = TRUE)

f4_a1 <- plot_go_bar(f4_a1_go_table, c(1:4, 63,65:67, 88:91), f4_pal) + 
  annotate("text", x = 12, y = 70, label = "bold('ATCC10231')", parse = TRUE)
```



```{r figure4, fig.cap=f4_cap, fig.width=7.2, fig.height=5}
# draw figure 4
ggarrange(f4_y, f4_a9, 
          f4_a1, f4_b, 
          labels = NULL, 
          ncol = 2, 
          nrow = 2, 
          common.legend = TRUE, 
          legend = "bottom", 
          align = "hv")

f4_cap <- "\\textbf{Functional enrichment analyses of the significantly 
           enriched and exclusive \\textit{C. albicans} EV proteins.} The 
           online tool FungiFun2 (elbe.hki-jena.de/fungifun) was used to 
           identify enriched biological process (BP), cellular component (CC), 
           and molecular function (MF) GO terms [@priebe2015]. A selection of 
           the most significantly enriched terms for each GO domain are shown. 
           They are presented top to bottom in order of increasing 
           Benjamini-Hochberg adjusted p-value. Full lists of enriched GO 
           terms can be found in supplemental File S3."
```

\newpage

```{r figure5Data, include=FALSE}
# figure 5a EV venn
f5a_comp <- list("DAY Y" = rownames(y_ev), 
                 "DAY B" = rownames(b_ev),
                 "ATCC90028" = rownames(atcc9_ev), 
                 "ATCC10231" = rownames(atcc1_ev))
f5a_comp <- lapply(f5a_comp, function(x) match_id(x, ref = uniprot, 
                                                  match = "UP_accession", 
                                                  new = "CGD_gene_name"))

f5a_venn <- plot_venn(vlist = f5a_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 1, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3))

f5a <- as_ggplot(f5a_venn) + 
  theme(plot.margin = unit(c(0,2,0,2), "lines"))

f5a_overlap <- plot_venn(vlist = f5a_comp, 
                         use_uniprot = FALSE, type = "df") %>% 
  filter(..count.. == 403) %>% 
  pull(..values..) %>% 
  unlist()

# figure 5b WCL venn
f5b_comp <- list("DAY Y" = rownames(y_wcl), 
                 "DAY B" = rownames(b_wcl),
                 "ATCC90028" = rownames(atcc9_w), 
                 "ATCC10231" = rownames(atcc1_w))
f5b_comp <- lapply(f5b_comp, function(x) match_id(x, ref = uniprot, 
                                                  match = "UP_accession", 
                                                  new = "CGD_gene_name"))

f5b_venn <- plot_venn(vlist = f5b_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 1, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3))

f5b <- as_ggplot(f5b_venn) + 
  theme(plot.margin = unit(c(0,2,0,2), "lines"))

# MDS data table
mds_data <- cbind(atcc_de, 
                  biofilm_de[match(rownames(atcc_de), rownames(biofilm_de)), ])

mds_data <- cbind(mds_data, 
                  yeast_de[match(rownames(mds_data), rownames(yeast_de)), ])

colnames(mds_data)[13:28] <- c("B_EV1", "B_EV2", "B_EV3", "B_EV4", "B_EV5", 
                               "B_W1", "B_W2", "B_W3", "B_W4", "B_W5", 
                               "Y_EV1", "Y_EV2", "Y_EV3", 
                               "Y_W1", "Y_W2", "Y_W3")

mds_labels <- c(rep("EV", 3), rep("WCL", 3), 
                rep("EV", 3), rep("WCL", 3), 
                rep("EV", 5), rep("WCL", 5), 
                rep("EV", 3), rep("WCL", 3))

# figure 5c MDS [dim1, dim2]
f5c <- plot_mds(mds_data, mat_labels = mds_labels, 
                shape_size = 3, dim.plot = c(1,2)) + 
  scale_x_continuous(limits = c(-1.5, 2.5), 
                     expand = c(0,0)) + 
  scale_y_continuous(limits = c(-1.5, 2.5), 
                     expand = c(0,0)) + 
  scale_fill_manual(breaks = c("Y", "A9", "A1", "B"), 
                    values = st_pal[c(3, 2, 4, 1)], 
                    labels = c("DAY Y", "A90028", "A10231", "DAY B"), 
                    guide = "legend") + 
  scale_shape_manual(values = c(22,24)) + 
  theme_bw() + 
  theme(legend.title = element_blank(),
        legend.position = "right", 
        legend.text = element_text(size = 8), 
        legend.direction = "vertical") + 
  guides(fill = guide_legend(override.aes = list(shape = 21))) + 
  xlab(label = "Dimension 1") + 
  ylab(label = "Dimension 2")
  
# figure 5d MDS [dim3, dim2]
f5d <- plot_mds(mds_data, mat_labels = mds_labels, 
                shape_size = 3, dim.plot = c(3,2)) + 
  scale_x_continuous(limits = c(-1.5, 2.5), 
                     expand = c(0,0)) + 
  scale_y_continuous(limits = c(-1.5, 2.5), 
                     expand = c(0,0)) + 
  scale_fill_manual(breaks = c("Y", "A9", "A1", "B"), 
                    values = st_pal[c(3, 2, 4, 1)], 
                    labels = c("DAY Y", "A90028", "A10231", "DAY B"), 
                    guide = "legend") + 
  scale_shape_manual(values = c(22,24)) + 
  theme_bw() + 
  theme(legend.title = element_blank(),
        legend.position = "right", 
        legend.text = element_text(size = 8), 
        legend.direction = "vertical", 
        legend.margin = margin(t = 0, r = 0, b = 0, l = 0.1), 
        legend.box.margin = margin(t = -10, r = -6, b = -10, l = -10)) + 
  guides(fill = guide_legend(override.aes = list(shape = 21))) + 
  xlab(label = "Dimension 3") + 
  ylab(label = "Dimension 2")
```

```{r figure5, fig.cap=f5_cap, fig.width=7.1, fig.height=5}
ggarrange(ggarrange(f5a, f5b,
                    labels = c("A", "B"),
                    ncol = 2,
                    nrow = 1),
          ggarrange(f5c, f5d,
                    labels = c("C", "D"),
                    ncol = 2,
                    nrow = 1, 
                    common.legend = TRUE,
                    legend = "right", 
                    align = "hv"),
          labels = NULL,
          ncol = 1,
          nrow = 2,
          heights = c(2, 3))

f5_cap <- "\\textbf{Combined comparison of \\textit{C. albicans} proteomes.} 
           \\textit{(A)} Venn diagram comparing four \\textit{C. albicans} EV 
           proteomes. Lists of proteins present in the centre and right 
           partitions of each venn diagram in Figure \\@ref(fig:figure3)A were 
           compared to identify proteins present in all four datasets. 
           \\textit{(B)} Venn diagram comparing four \\textit{C. albicans} WCL 
           proteomes. Lists of proteins present in the centre and left 
           partitions of each venn diagram in Figure \\@ref(fig:figure3)A were 
           compared to identify proteins present in all four datasets. The data 
           tables underlying Figures \\@ref(fig:figure5)A and 
           \\@ref(fig:figure5)B are provided in supplemental File S4. 
           \\textit{(C)} Multidimensional scaling (MDS) plot of each LC-MS/MS 
           biological replicate sample with the first 2 dimensions shown. 
           Pairwise distance between samples approximately indicates the 
           log\\textsubscript{2}(FC) between samples. \\textit{limma} was used 
           to generate the MDS plots [@ritchie2015]. \\textit{(D)} MDS plot of 
           each LC-MS/MS biological replicate sample with the second and third 
           dimensions shown."
```

\newpage

```{r figure6Heatmap, include=FALSE}
# figure 6 heatmap
f6_table <- a9_res %>% 
  select(logFC, CGD_gene_name) %>% 
  full_join(., select(a1_res, logFC, CGD_gene_name), 
            by = "CGD_gene_name") %>% 
  full_join(., select(yeast_res, logFC, CGD_gene_name), 
            by = "CGD_gene_name") %>% 
  full_join(., select(biofilm_res, logFC, CGD_gene_name), 
            by = "CGD_gene_name") %>% 
  rename(a9_lfc = logFC.x, 
         a1_lfc = logFC.y, 
         y_lfc = logFC.x.x, 
         b_lfc = logFC.y.y) %>% 
  select(CGD_gene_name, everything()) %>% 
  filter(CGD_gene_name %in% f5a_overlap)

# filter for rows with min 1/4 valid values
# otherwise clustering will not work
f6_table_filt <- f6_table %>% 
  filter_at(.vars = vars(contains("lfc")), 
            any_vars(!is.na(.))) %>% 
  tibble::column_to_rownames(var = "CGD_gene_name") %>% 
  as.matrix()

# excluded proteins
f6_table_excl <- f6_table %>% 
  filter_at(.vars = vars(contains("lfc")), 
            all_vars(is.na(.))) %>% 
  tibble::column_to_rownames(var = "CGD_gene_name") %>% 
  as.matrix()

# heatmap legend parameters
f6_params <- list(color_bar = "continuous", 
                  title = expression(bold(log[2]("FC"))), 
                  direction = "horizontal", 
                  legend_width = unit(2, "cm"), 
                  title_position = "lefttop", 
                  title_gp = grid::gpar(colour = "black", 
                                        fontsize = 5), 
                  labels_gp = grid::gpar(colour = "black", 
                                         fontsize = 5), 
                  at = seq(-6, 6, 2))

# plot heatmap
f6_htmp <- plot_heatmap(mt = f6_table_filt, 
                        plot = FALSE,
                        df = FALSE, 
                        data_type = "log2fc", 
                        clust_fun = "gower", 
                        split_type = "cutree", 
                        k = 5, 
                        cluster_split = FALSE, 
                        split_order = c(3, 1, 4, 2, 5), 
                        colour_lims = c(-4, 4),
                        column_labels = c("ATCC90028", "ATCC10231", 
                                          "DAY286 Y", "DAY286 B"), 
                        col_name_fontsize = 6, 
                        column_names_rot = 45, 
                        row_title_fontsize = 6, 
                        row_title_rot = 0, 
                        row_title = c("1", "2", "3", "4", "5"), 
                        heatmap_legend_param = f6_params)

# save heatmap as pdf file for editing in inkscape
# not run in rmarkdown but manually in console
# pdf(file = "figures/figure5_unf.pdf", width = 1.7, height = 4)
# ComplexHeatmap::draw(f5_htmp, heatmap_legend_side = "bottom")
# dev.off()
```

```{r figure6Table, include=FALSE}
#  list of proteins for heatmap table in order of appearance in heatmap
f6_clusters <- plot_heatmap(mt = f6_table_filt, 
                            plot = FALSE,
                            df = TRUE, 
                            data_type = "log2fc", 
                            clust_fun = "gower", 
                            split_type = "cutree", 
                            k = 5, 
                            cluster_split = FALSE) %>% 
  mutate(cluster = recode(cluster, 
                          "1" = 2, "2" = 4, "3" = 1, "4" = 3, "5" = 5)) %>% 
  arrange(cluster)

# extract identifiers from each cluster
f6_clust1 <- f6_clusters %>% 
  filter(cluster == 1) %>% 
  mutate(id = gsub("CHS3", "CaO19.4937", id)) %>% 
  pull(id)

f6_clust2 <- f6_clusters %>% 
  filter(cluster == 2) %>% 
  mutate(id = gsub("PEP1", "CaO19.3767", id)) %>% 
  pull(id) %>% 
  stringr::str_extract_all(., "[^;]+") %>% 
  unlist()
  
f6_clust3 <- f6_clusters %>% 
  filter(cluster == 3) %>% 
  pull(id)

f6_clust4 <- f6_clusters %>% 
  filter(cluster == 4) %>% 
  pull(id)

# cluster 5 has too many proteins to fit in table
# but GO enrichment still possible
f6_clust5 <- f6_clusters %>% 
  filter(cluster == 5) %>% 
  mutate(id = gsub("ALA1", "CaO19.5746", id), 
         id = gsub("MDH1", "CaO19.7481", id), 
         id = gsub("RPS1", "CaO19.3002", id), 
         id = gsub("TSA1", "CaO19.7417", id)) %>% 
  pull(id) %>% 
  stringr::str_extract_all(., "[^;]+") %>% 
  unlist() %>% 
  unique()

# cluster 6 is proteins that could not be clustered
# because all 4 log2FCs = NA
f6_clust6 <- sort(rownames(f6_table_excl))

# import FungiFun2 results
f6_go1 <- process_fungifun("figures/f6_GO_clust1.csv") %>% 
  mutate(Prot.found = gsub("CaO19.4937", "CHS3", Prot.found))

# issue with PEP1, ambiguous ID in FungiFun, used CaO19.3767 as identifier
f6_go2 <- process_fungifun("figures/f6_GO_clust2.csv") %>% 
  mutate(Prot.found = gsub("CaO19.3767", "PEP1", Prot.found))

f6_go3 <- process_fungifun("figures/f6_GO_clust3.csv")

# issue with CHS3, ambiguous ID in FungiFun, used CaO19.4937 as identifier
f6_go4 <- process_fungifun("figures/f6_GO_clust4.csv")

# issue with ALA1, MDH1, used CaO19.5746 and CaO19.7481 as identifiers
f6_go5 <- process_fungifun("figures/f6_GO_clust5.csv") %>% 
  mutate(Prot.found = gsub("CaO19.5746", "ALA1", Prot.found), 
         Prot.found = gsub("CaO19.7481", "MDH1", Prot.found), 
         Prot.found = gsub("CaO19.3002", "RPS1", Prot.found), 
         Prot.found = gsub("CaO19.7417", "TSA1", Prot.found))

f6_go6 <- process_fungifun("figures/f6_GO_clust6.csv")
```

```{r figure6, fig.cap=f6_cap, fig.height=6}
f6 <- png::readPNG("figures/figure6.png")

# draw figure 6
ggplot() + 
  background_image(f6) + 
  theme(plot.margin = margin(t = 0.1, l = 0.1, r = 0.1, b = 0.1, 
                             unit = "inches"))

f6_cap <- "\\textbf{Identification of commonly enriched EV proteins across 
           different \\textit{C. albicans} strains.} Heatmap of proteins 
           (n = 403) identified in all four EV samples (venn overlap in Figure 
           \\@ref(fig:figure5)A). Each column represents a 
           \\textit{C. albicans} strain and each row represents a protein. The 
           colours indicate the log\\textsubscript{2}(FC) of the protein for 
           that particular strain; red is EV enriched, blue is EV depleted, and 
           black is EV exclusive. Unsupervised clustering of rows and columns 
           was performed using Gower's formula via the daisy function from the 
           R package \\textit{cluster} [@maechler2019]. Proteins which were 
           exclusive to EVs from all four strains could not be clustered and 
           were seperated prior to clustering, in the group designated 
           \`\`cluster 6\". Functional enrichment analyses were performed on 
           the heatmap protein clusters using the online tool FungiFun2 
           [@priebe2015]. A selection of significantly enriched GO terms are 
           presented in the table inset. Lists of the proteins identified in 
           each cluster are shown and are also available in supplemental 
           File S5."
```

\newpage

```{r figure7Data, include=FALSE}
f7a_venn <- plot_venn(vlist = marker_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 0.9, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3), 
                      sub = "EV exclusive + sig. enriched", 
                      sub.fontfamily = "sans", 
                      sub.pos = c(0.5, 1.05), 
                      margin = 0.05)

f7a <- as_ggplot(f7a_venn) + 
  theme(plot.margin = unit(c(1.2,0,0,0), "lines"))

wcl_comp <- list("DAY Y" = yeast_res %>% 
                   filter(group == "WCL up" | group == "WCL ex") %>% 
                   pull(CGD_gene_name), 
                 "DAY B" = biofilm_res %>% 
                   filter(group == "WCL up" | group == "WCL ex") %>% 
                   pull(CGD_gene_name),
                 "ATCC90028" = a9_res %>% 
                   filter(group == "A9_W up" | group == "A9_W ex") %>% 
                   pull(CGD_gene_name), 
                 "ATCC10231" = a1_res %>% 
                   filter(group == "A1_W up" | group == "A1_W ex") %>% 
                   pull(CGD_gene_name))

f7b_venn <- plot_venn(vlist = wcl_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 0.9, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3), 
                      sub = "WCL exclusive + sig. enriched", 
                      sub.fontfamily = "sans", 
                      sub.pos = c(0.5, 1.05), 
                      margin = 0.05)

f7b <- as_ggplot(f7b_venn) + 
  theme(plot.margin = unit(c(1.2,0,0,0), "lines"))
```

```{r figure7, fig.cap=f7_cap, fig.width=2.5, fig.height=4}
ggarrange(f7a, f7b, 
          labels = c("A", "B"), 
          ncol = 1,
          nrow = 2)

f7_cap <- "\\textbf{Identification of common signficantly enriched or exclusive 
           \\textit{C. albicans} EV and WCL proteins.} Proteins identified as 
           significantly enriched or exclusive to EVs in Figure 
           \\@ref(fig:figure3) were compared to select those which were common 
           to all 4 datasets. 43 proteins were identified as potential positive 
           EV protein markers and 52 proteins as potential negative EV protein 
           markers. The data tables underlying Figure 
           \\@ref(fig:figure7) are provided in supplemental File S4. Details of 
           the 43 positive EV protein marker candidates and 52 negative 
           candidates can be found in Table \\@ref(tab:table1) and supplemental 
           Table S2 respectively."
```

\newpage

```{r figure8, fig.cap=f8_cap, fig.width=3.5, fig.height=2.8}
f8_cap <- "\\textbf{Eighteen best potential \\textit{C. albicans} positive EV 
           protein markers.} These proteins are those underlined in Table 
           \\@ref(tab:table1). They are are enriched or exclusive to EVs 
           isolated from all of the \\textit{C. albicans} strains investigated 
           in this study. The proteins fit within the EV biomarker categories 
           1a, 2a, and 2b from MISEV2018 which includes transmembrane, 
           GPI-anchored, and cytosolic proteins that are EV enriched 
           [@thery2018]."

f8 <- png::readPNG("figures/figure8.png")

ggplot() + 
  background_image(f8)
```

\newpage

```{r figure9, fig.cap=f9_cap, fig.width=7.2, fig.height=6}
f9_cap <- "\\textbf{Predicted topology and palmitoylation sites of 
           \\textit{C. albicans} MCC/eisosome proteins SUR7 and orf19.6741 
           compared to the human tetraspanin CD81.} The presence of 
           transmembrane (TM) domains and palmitoylation sites shown on the 
           left side of the figure were predicted using the online tools 
           TOPCONS2 (topcons.cbr.su.se) [@tsirigos2015] and CSS-PALM 4.0 
           (csspalm.biocuckoo.org) [@ren2008] respectively. For CD81, 
           palmitoylation sites experimentally determined in previous 
           literature are shown [@delandre2009]. The right side of the figure 
           shows a cartoon representation of the TOPCONS2 consensus topology 
           prediction for the three proteins as well as the crystal structure 
           of CD81 (PDB: 5TCX) [@zimmerman2016]."

f9 <- png::readPNG("figures/figure9.png")

ggplot() + 
  background_image(f9)
```

