---
title: "**Defining protein biomarkers for *Candida albicans* extracellular vesicles**"
author: "*Charlotte Dawson^a^, Mark Bleackley^a^, Marilyn Anderson^a^*"
date: "*^a^Department of Biochemistry and Genetics, La Trobe University, Victoria, Australia*"
output: 
    bookdown::pdf_document2:
      latex_engine: xelatex
      fig_caption: yes
      number_sections: yes
      toc: no
      keep_tex: no
    bookdown::word_document2:
      pandoc_args:
        '--lua-filter=components/pagebreak.lua'
      reference_docx: "components/draft_template.docx"
      fig_caption: yes
bibliography: "components/references.bib" 
csl: "components/molecular-and-cellular-proteomics.csl"
link-citations: yes
linkcolor: blue
fontsize: 12pt
geometry: margin=1in
classoption: letterpaper
mainfont: Times New Roman
linestretch: 1.5
header-includes:
  - \usepackage{titling}
  - \pretitle{\begin{center}\fontsize{14bp}{14bp}\selectfont}
  - \posttitle{\par\end{center}}
  - \preauthor{\begin{center}\fontsize{12bp}{12bp}\selectfont}
  - \postauthor{\par\end{center}}
  - \predate{\begin{center}\fontsize{10bp}{10bp}\selectfont}
  - \postdate{\par\end{center}}
  - \usepackage{fancyhdr}
  - \usepackage{lipsum}
  - \pagestyle{fancy}
  - \fancyhead[L]{This is the running title it is rather long}
  - \setlength{\headheight}{15pt}
  - \usepackage{float}
  - \let\origfigure\figure
  - \let\endorigfigure\endfigure
  - \renewenvironment{figure}[1][2]{\expandafter\origfigure\expandafter[H]}{\endorigfigure}
  - \usepackage{caption}
  - \captionsetup{font={normalsize,stretch=1.3},labelfont=bf}
  - \usepackage{threeparttable}
---

**Running title:** Running title goes here

**Corresponding author:**

\newpage

<!-- This is the format for text comments that will be ignored during renderings. Do not put R code in these comments because it will not be ignored. -->

```{r package_options, include=FALSE}
knitr::opts_knit$set(progress = TRUE, 
                     verbose = TRUE)
```

```{r setup, echo = FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(collapse = TRUE,
                      warning = FALSE,
                      message = FALSE,
                      echo = FALSE,
                      comment = "#>")

library(candidaev)
library(tidyNano)
library(multipanelfigure)
library(limma)
library(dplyr)
library(ggplot2)
library(candidaev)
library(ggpubr)
library(UpSetR)
library(kableExtra)
set.seed(1)

# define colour palette to distinguish between the 3 Candida strains and 
# 2 morphologies
st_pal <- c(y = "#E69F00", b =  "#CC79A7", a9 =  "#56B4E9", a1 = "#009E73")

# define colour palette to distinguish between EVs and whole cell lysate (WCL)
ev_pal <- c("#999999", "#D55E00", "#0072B2")

```

```{r yeast, warning=FALSE}
yeast <- yeast %>% 
  filter(Reverse != "+", Potential.contaminant != "+", 
         Only.identified.by.site != "+", Unique.peptides >= 2)

yeast_lfq <- convert_lfq(yeast, yeast_exp)

# drop EV_1 and WCL_2
yeast_lfq2 <- yeast_lfq[, !colnames(yeast_lfq) %in% c("EV_1", "WCL_2")]

# filter for proteins identified in min 2/3 reps of EV or WCL
yeast_filt2 <- filter_na2(yeast_lfq2, logic = "or", op = "<=", 
                          pat1 = "EV", val1 = 1, 
                          pat2 = "W", val2 = 1)

yeast_norm2 <- normalizeCyclicLoess(yeast_filt2)

# filter for proteins identified in min 1/3 reps of EV
y_ev <- filter_na(yeast_norm2, op = "<=", 
                  pat = "EV", val = 2)

# filter for proteins identified in min 1/3 reps of WCL
y_wcl <- filter_na(yeast_norm2, op = "<=", 
                   pat = "W", val = 2)

# should not impute with so many missing values
yeast_excl <- filter_na2(yeast_norm2, logic = "or", op = "==", 
                         pat1 = "EV", val1 = 3, 
                         pat2 = "W", val2 = 3)

yeast_both <- filter_na2(yeast_norm2, logic = "and", op = "<=", 
                         pat1 = "EV", val1 = 2, 
                         pat2 = "W", val2 = 2)

yeast_imp <- impute_QRILC(yeast_both)

# recombine imputed proteins and non-imputed proteins in matrix
yeast_de <- rbind(yeast_excl, yeast_imp)

# see limma user guide section 9.2 for more info about DE
# create design matrix
y_samp <- data.frame(T = (rep(c("EV", "WCL"), each = 3)))

y_design <- stats::model.matrix(~ 0 + T, data = y_samp)
colnames(y_design) <- c("EV", "WCL")

y_contrasts <- c("EV - WCL")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
y_efit <- limma_eBayes(yeast_de, design = y_design, contrasts = y_contrasts)

# extract DE results
yeast_res <- get_results(efit = y_efit, mat = yeast_de, p_val = 0.01, lfc = 0, type = "individual")[[1]]

```

```{r biofilm, warning=FALSE}
biofilm <- biofilm %>% 
  filter(Reverse != "+", Potential.contaminant != "+", 
         Only.identified.by.site != "+", Unique.peptides >= 2)

biofilm_lfq <- convert_lfq(biofilm, biofilm_exp)

# filter for proteins identified in min 4/5 reps of EV or WCL
biofilm_filt <- filter_na2(biofilm_lfq, logic = "or", op = "<=", 
                           pat1 = "EV", val1 = 1, 
                           pat2 = "W", val2 = 1)

biofilm_norm <- normalizeCyclicLoess(biofilm_filt)

# filter for proteins identified in min 1/5 reps of EV
b_ev <- filter_na(biofilm_norm, op = "<=", 
                  pat = "EV", val = 4)

# filter for proteins identified in min 1/5 reps of WCL
b_wcl <- filter_na(biofilm_norm, op = "<=", 
                   pat = "W", val = 4)

# should not impute with so many missing values
# filter for proteins with 4-5 NA values in EV or WCL
biofilm_excl <- filter_na2(biofilm_norm, logic = "or", op = ">=", 
                         pat1 = "EV", val1 = 4, 
                         pat2 = "W", val2 = 4)

# filter for proteins with min 3 valid values in EV and WCL
biofilm_both <- filter_na2(biofilm_norm, logic = "and", op = "<=", 
                         pat1 = "EV", val1 = 3, 
                         pat2 = "W", val2 = 3)

# proteins with missing values tend to have lower intensity
# therefore proteins are MNAR, close to detection limit

# use left censored imputation method
biofilm_imp <- impute_QRILC(biofilm_both)

# recombine imputed proteins and non-imputed proteins in matrix
biofilm_de <- rbind(biofilm_excl, biofilm_imp)

# see limma user guide section 9.2 for more info about DE
# create design matrix
b_samp <- data.frame(T = (rep(c("EV", "WCL"), each = 5)))

b_design <- stats::model.matrix(~ 0 + T, data = b_samp)
colnames(b_design) <- c("EV", "WCL")

b_contrasts <- c("EV - WCL")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
b_efit <- limma_eBayes(biofilm_de, design = b_design, contrasts = b_contrasts)

# extract DE results
biofilm_res <- get_results(efit = b_efit, mat = biofilm_de, p_val = 0.01, lfc = 0, type = "individual")[[1]]

```

```{r atcc, warning=FALSE}
atcc_lfq <- atcc %>% 
  filter(Reverse != "+", Potential.contaminant != "+", Unique.peptides >= 2) %>% 
  convert_lfq(., atcc_exp)

atcc_filt <- filter_na4(atcc_lfq, "or", "<=", 
                        pat1 = "A10231_EV", 1, 
                        pat2 = "A10231_W", 1, 
                        pat3 = "A90028_EV", 1, 
                        pat4 = "A90028_W", 1)

atcc_norm <- normalizeCyclicLoess(atcc_filt)

# filter for proteins identified in min 1/3 reps of 10231 EV
atcc1_ev <- filter_na(atcc_norm, op = "<=", 
                      pat = "A10231_EV", val = 2)

# filter for proteins identified in min 1/3 reps of 10231 WCL
atcc1_w <- filter_na(atcc_norm, op = "<=", 
                     pat = "A10231_W", val = 2)

# filter for proteins identified in min 1/3 reps of 90028 EV
atcc9_ev <- filter_na(atcc_norm, op = "<=", 
                      pat = "A90028_EV", val = 2)

# filter for proteins identified in min 1/3 reps of 90028 WCL
atcc9_w <- filter_na(atcc_norm, op = "<=", 
                      pat = "A90028_W", val = 2)

# should not impute with so many missing values
# filter for exclusive to 10231 EV or WCL or 90028 EV or WCL
atcc_excl <- filter_na4(atcc_norm, logic = "or", op = "==", 
                        pat1 = "A10231_EV", val1 = 3, 
                        pat2 = "A10231_W", val2 = 3, 
                        pat3 = "A90028_EV", val3 = 3, 
                        pat4 = "A90028_W", val4 = 3)

# filter for proteins with min 1 valid value in all 4 sample types
atcc_both <- filter_na4(atcc_norm, logic = "and", op = "<=", 
                        pat1 = "A10231_EV", val1 = 2, 
                        pat2 = "A10231_W", val2 = 2, 
                        pat3 = "A90028_EV", val3 = 2, 
                        pat4 = "A90028_W", val4 = 2)

# proteins with missing values tend to have lower intensity
# therefore proteins are MNAR, close to detection limit

# use left censored imputation method
atcc_imp <- impute_QRILC(atcc_both)

atcc_de <- rbind(atcc_excl, atcc_imp)

colnames(atcc_de) <- stringi::stri_replace_all_regex(colnames(atcc_de), 
                                                     c("A10231_", "A90028_"), 
                                                     c("A1_", "A9_"), 
                                                     vectorize_all = FALSE)

# see limma user guide section 9.2 for more info about DE
# create design matrix
atcc_samp <- data.frame(T = (rep(c("A1_EV", "A1_WCL", "A9_EV", "A9_WCL"), each = 3)))

atcc_design <- stats::model.matrix(~ 0 + T, data = atcc_samp)
colnames(atcc_design) <- c("A1_EV", "A1_W", "A9_EV", "A9_W")

atcc_contrasts <- c("A1_EV - A1_W", "A9_EV - A9_W", "A1_EV - A9_EV", "A1_W - A9_W")

# make all pair-wise comparisons for specified contrasts
# and perform limma::eBayes()
atcc_efit <- limma_eBayes(atcc_de, atcc_design, atcc_contrasts)

# extract overall results
atcc_overall <- get_results(efit = atcc_efit, mat = atcc_de, p_val = 0.01, lfc = 0, type = "overall")

# extract results for individual comparisons defined in atcc_contrasts
atcc_ind <- get_results(efit = atcc_efit, mat = atcc_de, p_val = 0.01, lfc = 0, type = "individual")

a1_res <- atcc_ind[[1]]

a9_res <- atcc_ind[[2]]

```


# Abbreviations

EV - extracellular vesicle

\newpage

# Abstract

The abstract goes here. Here is a citation [@liao2012].

Here is another citation [@gil-bona2015].

And another [@lin2014].

\newpage

# Introduction

Here is a citation [@marwick2017].

# Experimental procedures

## Subsection 1

## Subsection 2

# Results

# Discussion

# Acknowledgements

# Data availability

# Acknowledgements

\newpage
# References 
<!-- The following line ensures the references appear here for the MS Word or HTML output files, rather than right at the end of the document (this will not work for PDF files):  -->
<div id="refs"></div>

\newpage

# Tables

```{r table1Data}
marker_comp <- list("yeast" = yeast_res %>% 
                      filter(group == "EV up" | group == "EV ex") %>% 
                      pull(CGD_gene_name), 
                    "biofilm" = biofilm_res %>% 
                      filter(group == "EV up" | group == "EV ex") %>% 
                      pull(CGD_gene_name), 
                    "ATCC10231" = a1_res %>% 
                      filter(group == "A1_EV up" | group == "A1_EV ex") %>% 
                      pull(CGD_gene_name), 
                    "ATCC90028" = a9_res %>% 
                      filter(group == "A9_EV up" | group == "A9_EV ex") %>% 
                      pull(CGD_gene_name))

marker_comp_tab <- plot_venn(vlist = marker_comp, use_uniprot = FALSE, type = "list")

marker_candidates <- marker_comp_tab %>% 
  filter(apply(.[, 1:4], 1, sum) == 4) %>% 
  pull(..values..) %>% 
  unlist() %>% 
  as.data.frame(stringsAsFactors = FALSE) %>% 
  rename(Name = 1)

marker_longlist <- marker_candidates %>% 
  mutate(Function = match_id(Name, uniprot, "CGD_gene_name", "Protein_name"),
         DAY.Y = match_id(Name, yeast_res, "CGD_gene_name", "logFC", str_split = FALSE), 
         DAY.B = match_id(Name, biofilm_res, "CGD_gene_name", "logFC", str_split = FALSE), 
         A1 = match_id(Name, a1_res, "CGD_gene_name", "logFC", str_split = FALSE), 
         A9 = match_id(Name, a9_res, "CGD_gene_name", "logFC", str_split = FALSE), 
         TM = match_id(Name, uniprot, "CGD_gene_name", "Transmembrane"), 
         SP = match_id(Name, uniprot, "CGD_gene_name", "Signal_peptide")) %>% 
  mutate_at(.vars = vars(DAY.Y, DAY.B, A1, A9), 
            .funs = list(~ signif(as.numeric(.), digits = 3))) %>% 
  mutate_at(.vars = vars(DAY.Y, DAY.B, A1, A9), 
            .funs = list(~ case_when(is.na(.) == TRUE ~ "ex", 
                                     TRUE ~ as.character(.)))) %>% 
  mutate(TM = stringr::str_count(TM, "Helical"), 
         SP = case_when(grepl("SIGNAL", SP) ~ "Y", 
                        TRUE ~ ""), 
         TM = case_when(TM == 0 ~ "", 
                        TRUE ~ as.character(TM))) %>% 
  rename("DAY Y" = DAY.Y, "DAY B" = DAY.B) %>% 
  arrange(Name)

protein_desc <- c("1,3-beta-glucanosyltransferase", 
                  "Rho family GTPase", 
                  "Multidrug transporter of ABC superfamily", 
                  "Major chitin synthase of yeast and hyphae", 
                  "GPI-anchored cell wall transglycosylase", 
                  "GPI-anchored cell wall protein", 
                  paste0("Predicted P-type ATPase sodium pump", footnote_marker_alphabet(1)), 
                  "ER oxidoreductin", 
                  paste0("Long-chain fatty acid-CoA ligase", footnote_marker_alphabet(1)),
                  "Multicopper feroxidase", 
                  "High-affinity S-adenosylmethionine permease", 
                  "Surface protein similar to glycerol-3-phosphate dehydrogenase", 
                  "1,3-beta-glucan synthase", 
                  "High-affinity MFS glucose transporter", 
                  "Putative high-affinity MFS glucose transporter", 
                  paste0("Putative mitochondrial phosphate transporter", footnote_marker_alphabet(1)), 
                  paste0("Glycolipid 2-alpha-mannosyltransferase", footnote_marker_alphabet(1)), 
                  "Cell surface mannoprotein", 
                  "Mucin family adhesin-like protein", 
                  "Sphingolipid C9-methyltransferase", 
                  paste0("\\textit{S. cerevisiae} ortholog is POM33, transmembrane nucleoporin", footnote_marker_alphabet(2)), 
                  paste0("\\textit{S. cerevisiae} ortholog is RTN1, reticulon protein", footnote_marker_alphabet(2)), 
                  paste0("Hexadecenal dehydrogenase", footnote_marker_alphabet(2)), 
                  paste0("\\textit{S. cerevisiae} ortholog is PUN1, plasma membrane protein", footnote_marker_alphabet(2)), 
                  "1,3-beta-glucanosyltransferase", 
                  "GPI-anchored cell surface protein of unknown function", 
                  "Putative ion transporter", 
                  "Cell surface glycosidase", 
                  "Glycosidase", 
                  "Phospholipase B", 
                  "Plasma membrane ATPase", 
                  paste0("Mitochondrial outer membrane porin", footnote_marker_alphabet(1)), 
                  "G-protein of RAC subfamily", 
                  "Rho family GTPase", 
                  "Rho family GTPase", 
                  "Secreted aspartyl protease", 
                  "Rab family GTPase", 
                  paste0("Protein required for normal cell wall, plasma membrane", footnote_marker_alphabet(3)), 
                  paste0("Protein involved in vacuolar inheritance", footnote_marker_alphabet(1)), 
                  "\\textit{S. cerevisiae} ortholog is YCK2, casein kinase", 
                  paste0("Palmitoyltransferase, putative vacuolar SNARE complex protein", footnote_marker_alphabet(1)), 
                  paste0("Rab family GTPase", footnote_marker_alphabet(1)),  
                  "Putative vacuolar cation channel")
marker_longlist <- marker_longlist %>% 
  mutate(Function = protein_desc, 
         VDS = c("Y", rep("",3), rep("Y", 2), "", "Y", rep("", 9), rep("Y", 2), 
                 rep("", 4), rep("Y", 3), rep("", 2), rep("Y", 2), rep("", 5), "Y", rep("", 7)), 
         pack_row = c(5, rep(1,3), rep(5,2), 1, 2, rep(1,3), 5, rep(1,3), 4, 2, rep(5,2), 
                      1, rep(2,2), 4, 1, rep(5,2), 1, rep(5,3), 1, 4, rep(1,3), 5, 2, 1, 
                      3, 1, 3, 2, 3)) %>% 
  arrange(pack_row)

# fix number of TM domains for CDR1;CDR2
marker_longlist[2, 7] <- "12;12" 

# UniProt says SUR7 has 3 TM domains and a signal peptide
# Much of literature says SUR7 has 4 TM domains
# TOPCONS2 predicts SUR7 has 4 TM domains and no signal peptide
# fix SUR7 entries
marker_longlist[18, 7] <- "4"
marker_longlist[18, 8] <- ""

                     
```

(ref:table1) [@gil-bona2015]

(ref:footnote1) [@skrzypek2017]

(ref:footnote2) [@almagroarmenteros2017]

(ref:footnote3) [@tsirigos2015]

```{r table1}
tab1_cap <- "\\textbf{Candidate positive protein markers for \\textit{C. albicans} EVs.} 
             This list of proteins consists of those that were found to be exclusive to EVs or 
             significantly enriched in EVs across the four \\textit{C. albicans} strains examined in 
             this study. Proteins are grouped according to their subcellular localisation as annotated 
             in the Candida Genome Database (candidagenome.org) (ref:footnote1). The log\\textsubscript{2} ratio 
             of the abundance (mean MaxQuant LFQ intensity) of each protein in EVs compared to 
             whole cell lysate for each strain is listed. \`\`ex\" indicates where a protein was only 
             identified in the EV fraction and not the whole cell lysate for that strain. The \`\`TM\" 
             column indicates the number of transmembrane domains for each protein as annotated in 
             UniProtKB. \`\`SP\" indicates whether a protein is annotated as having a signal peptide 
             according to UniProtKB. \`\`VDS\" shows whether a protein has previously been detected in 
             vesicle-depleted culture media (i.e. the proteins may also be in the soluble secretome) 
             (ref:table1). Underlined proteins are those identified as the best candidates for positive
             EV markers according to the criteria depicted in Figure \\ref{fig:figure1}."

knitr::kable(marker_longlist[ ,-10],
             caption = tab1_cap, 
             format = "latex", 
             booktabs = TRUE, 
             longtable = TRUE, 
             linesep = "", 
             escape = FALSE)  %>% 
  row_spec(row = c(1, 5, 6, 11, 12, 15:19, 21:27, 35, 40), underline = TRUE) %>% 
  add_header_above(c(" " = 2, "log\\\\textsubscript{2}(fold change) EV vs WCL" = 4, " " = 2), escape = FALSE) %>% 
  kable_styling(font_size = 8) %>% 
  pack_rows("Plasma membrane", 1, 19) %>% 
  pack_rows("Golgi, ER, secretory pathway", 20, 25) %>% 
  pack_rows("Vacuole", 26, 28) %>% 
  pack_rows("Mitochondria", 29, 31) %>% 
  pack_rows("Cell wall, cell surface", 32, 43) %>% 
  footnote(alphabet = c("Protein localisation was inferred from sequence similarity with 
                         \\\\textit{S. cerevisiae} ortholog as annotated in the Candida 
                         Genome Database (ref:footnote1)", 
                        "Protein localisation was predicted using DeepLoc-1.0 (ref:footnote2)", 
                        "Presence of transmembrane domains and absense of a signal peptide was predicted 
                         using TOPCONS2 (ref:footnote3)"), 
           escape = FALSE, 
           threeparttable = TRUE)

```

\newpage

# Figures

```{r figure1, fig.cap=f1_cap, fig.width=5.45, fig.height=6}

f1_cap <- "\\textbf{Schematic of the workflow used for the isolation and proteomic analysis of 
           \\textit{C. albicans} EVs and the selection of EV protein markers.} Steps 1 to 6 
           were performed for each of the four strains examined in this study: DAY286 yeast, 
           ATCC90028 yeast, ATCC10231 yeast, and DAY286 biofilm. 
           The results from the four seperate analyses were combined (step 7) to identify 
           proteins which commonly appeared as EV marker candidates. The MISEV2018 criteria 
           for protein content-based EV characterisation and other considerations were 
           used to select the best \\textit{C. albicans} EV marker proteins."

f1 <- png::readPNG("./vignettes/figures/figure1.png")

ggplot() + 
  background_image(f1)
```

\newpage

```{r figure2Data}
nta_metadata <- data.frame(id = c("ATCC10231_1", "ATCC10231_2", "ATCC10231_3", 
                                  "ATCC90028_1", "ATCC90028_2", "ATCC90028_3", 
                                  "Biofilm_1", "Biofilm_2", "Biofilm_3", "Biofilm_4", "Biofilm_5", 
                                  "DAY286_1", "DAY286_2", "DAY286_3"), 
                           protein_conc = c(0.192, 0.282, 0.287, 
                                            0.424, 0.716, 0.584, 
                                            0.204, 0.198, 0.297, 0.193, 0.416, 
                                            0.437, 0.360, 0.447), 
                           stringsAsFactors = FALSE)

nta_techavg <- nta %>% 
  nanolyze(particle_size, Strain, Bio_rep, Dilution, 
           name = "Tech_rep", 
           param_var = True_count)

nta_bioavg <- nta_techavg %>% 
  nanolyze(particle_size, Strain,
           name = "Bio_rep", 
           param_var = Tech_rep_mean) %>% 
  mutate(facet = case_when(Strain == "Biofilm" ~ "first", TRUE ~ "second"))

nta_total <- nta_techavg %>% 
  nanocount(Strain, Bio_rep, Dilution, 
            param_var = Tech_rep_mean) %>% 
  tidyr::unite("id", Strain, Bio_rep) %>% 
  mutate(protein_conc = as.numeric(match_id(id, nta_metadata, "id", "protein_conc")), 
         od = match_id(id, nta_metadata, "id", "OD600"), 
         part_per_prot = Particle_count / protein_conc / 1000) %>% 
  tidyr::separate(col = id, into = c("Strain", "Bio_rep"), sep = "_", remove = TRUE)

nta_total_tukey <- aov(part_per_prot ~ Strain, data = nta_total) %>% 
  tukey_hsd()

nta_total_dunns <- dunns_test(x = nta_total$Particle_count, 
                              g = nta_total$Strain, 
                              method = "bh")

```


```{r figure2plots}
f2a_img <- png::readPNG("./vignettes/figures/figure2a.png")

f2a <- ggplot() + 
  background_image(f2a_img) + 
  theme(plot.margin = margin(t = 0.3, l = 0, r = 0, b = 0, unit = "inches"))

f2b <- nta_bioavg %>% 
  ggplot(aes(x = particle_size, y = Bio_rep_mean, colour = Strain)) + 
  scale_colour_manual(values = rev(unname(st_pal)), 
                      breaks = c("Biofilm", "DAY286", "ATCC90028", "ATCC10231"), 
                      labels = c("DAY286 B", "DAY286 Y", "ATCC90028", "ATCC10231")) + 
  geom_vline(xintercept = 100, colour = "#999999", linetype = "dashed") +
  geom_line(size = 0.5) + 
  facet_wrap(~ facet, scales = "free", ncol = 2, nrow = 1) + 
  theme_classic() + 
  theme(strip.background = element_blank(), 
        strip.text = element_blank(), 
        legend.position = c(0.88, 0.6), 
        legend.title = element_blank(), 
        plot.margin = margin(t = 0.2, l = 0, r = 0.2, b = 0, unit = "inches"), 
        axis.text.x = element_text(colour = "black")) + 
  xlab(label = "Particle size (nm)") + 
  ylab(label = "Particles / mL")

f2c1 <- nta_total %>% 
  ggboxplot(x = "Strain", y = "Particle_count", 
            color = "Strain", palette = "jco", add = "jitter")
f2c1 <- ggpar(f2c1, 
              yscale = "log10", 
              format.scale = TRUE,
              ylim = c(1e11, 1e13)) + 
  stat_pvalue_manual(nta_total_dunns[c(1,2), ], 
                     label = "p.adj", 
                     y.position = c(12.4, 13), 
                     tip.length = 4e-3)

f2c2 <- nta_total %>% 
  ggboxplot(x = "Strain", y = "part_per_prot", 
            color = "Strain", palette = rev(unname(st_pal)), add = "jitter", 
            shape = 15,
            ylab = expression("Particles /"~paste(mu,g)~"protein"))
f2c2 <- ggpar(f2c2, 
              yscale = "log10", 
              format.scale = TRUE, 
              ylim = c(8e8, 8e10),
              font.y = 10, 
              legend = "none") + 
  stat_pvalue_manual(nta_total_tukey[c(2,4,6), ], label = "p.adj", 
                     y.position = c(10.7, 10.5, 10.85), 
                     tip.length = 1e-2, 
                     label.size = 3) + 
  theme(axis.title.x = element_blank(), 
        plot.margin = margin(t = 0.2, l = 0.2, r = 0, b = 0.15, unit = "inches")) + 
  scale_x_discrete(labels = c("ATCC10231" = "A1", "ATCC90028" = "A9", "Biofilm" = "B", "DAY286" = "Y"))

```


```{r figure2, fig.cap=f2_cap, fig.width=7.1, fig.height=4}

ggarrange(f2a, 
          ggarrange(f2b, f2c2, 
                    labels = c("B", "C"), 
                    ncol = 2,
                    nrow = 1, 
                    widths = c(1.8,1)), 
          labels = c("A", ""), 
          nrow = 2, 
          ncol = 1, 
          heights = c(1,1))

f2_cap <- "\\textbf{Characterisation and quantification of EVs from different \\textit{C. albicans} strains.} 
           \\textit{(A)} Representative TEM images of EVs isolated from 4 different \\textit{C. albicans} cultures.
           Scale bar indicates 0.5 $\\mu$m. \\textit{(B)} Size distribution of DAY286 biofilm (n = 5), DAY286 yeast (n = 3), 
           ATCC90028 yeast (n = 3), and ATCC10231 yeast (n = 3) EVs as measured by nanoparticle tracking analysis. 
           Line plots shown are the average of the biological replicates and the dashed grey line indicates 100 nm. 
           Plots for the individual biological replicates are presented in supplemental Figure S1.
           \\textit{(C)} Comparison of the ratio of particle concentration to protein concentration of the EV fraction, 
           (particles/mL) $\\div$ ($\\mu$g protein/mL), across the four EV sources. 
           Each dot represents one biological replicate. Sample means were compared using ANOVA followed by 
           Tukey's HSD \\textit{post-hoc} test. P-values indicating significant 
           differences are shown. The particle and protein concentrations for each biological replicate are 
           provided in supplemental Table S1."

```

\newpage

```{r figure3aData}
f3a1 <- plot_venn(list("EV" = y_ev, "WCL" = y_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8,
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  margin = 0.05)

f3a1 <- as_ggplot(f3a1) + 
  theme(plot.margin = unit(c(1,0.5,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('DAY286 yeast')", parse = TRUE)

f3a2 <- plot_volcano(data = yeast_res, 
                     p_val = adj.P.Val, 
                     lfc = logFC,
                     grp = group, 
                     label = ifelse(logFC >= 8, CGD_gene_name, "")) + 
  scale_colour_manual(labels = c("EV enriched", "WCL enriched", "Not significant"), 
                      values = unname(ev_pal[c(2,1,3)]), 
                      breaks = c("EV up", "WCL up", "not sig"), 
                      name = element_blank()) + 
  scale_x_continuous(limits = c(-10,12), expand = c(0,0)) + 
  scale_y_continuous(limits = c(0, 8), expand = c(0,0)) + 
  theme(legend.text = element_text(size = 8)) + 
  annotate("text", x = 8.5, y = 0.5, label = "bold('DAY286 Y')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('80')", parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('109')", parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('318')", parse = TRUE, colour = unname(ev_pal[1]))


```



```{r figure3bData}
f3b1 <- plot_venn(list("EV" = b_ev, "WCL" = b_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8, 
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  ext.text = TRUE,
                  ext.percent = c(20,10,0),
                  ext.pos = c(45,315),
                  ext.dist = -0.1,
                  ext.length = 0.5,
                  margin = 0.05)

f3b1 <- as_ggplot(f3b1) + 
  theme(plot.margin = unit(c(1,0,1,0), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('DAY286 biofilm')", parse = TRUE)

f3b2 <- plot_volcano(data = biofilm_res, 
                     p_val = adj.P.Val, 
                     lfc = logFC,
                     grp = group, 
                     label = ifelse(logFC >= 8, CGD_gene_name, "")) + 
  scale_colour_manual(labels = c("EV enriched (95)", "WCL enriched (81)", "Not significant (482)"), 
                      values = unname(ev_pal[c(2,1,3)]), 
                      breaks = c("EV up", "WCL up", "not sig")) + 
  scale_x_continuous(limits = c(-10,12), expand = c(0,0)) + 
  scale_y_continuous(limits = c(0, 8), expand = c(0,0)) + 
  theme(legend.text = element_text(size = 8)) + 
  annotate("text", x = 8.5, y = 0.5, label = "bold('DAY286 B')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('95')", parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('81')", parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('482')", parse = TRUE, colour = unname(ev_pal[1]))



```



```{r figure3cData}
f3c1 <- plot_venn(list("EV" = atcc9_ev, "WCL" = atcc9_w), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans",
                  cex = 0.8,
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.55)), 
                  ext.percent = c(30,10,0),
                  ext.pos = c(45,315),
                  ext.dist = -0.05,
                  ext.length = 0.5,
                  margin = 0.05)

f3c1 <- as_ggplot(f3c1) + 
  theme(plot.margin = unit(c(1,0,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('ATCC90028')", parse = TRUE)

f3c2 <- plot_volcano(data = a9_res, 
                     p_val = adj.P.Val, 
                     lfc = logFC,
                     grp = group, 
                     label = ifelse(logFC >= 8, CGD_gene_name, "")) + 
  scale_colour_manual(labels = c("EV enriched (105)", "WCL enriched (134)", "Not significant (734)"),
                      values = unname(ev_pal[c(2,3,1)])) + 
  scale_x_continuous(limits = c(-10,12), expand = c(0,0)) + 
  scale_y_continuous(limits = c(0, 8), expand = c(0,0)) + 
  theme(legend.text = element_text(size = 8)) + 
  annotate("text", x = 8, y = 0.5, label = "bold('ATCC90028')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('105')", parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('134')", parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('734')", parse = TRUE, colour = unname(ev_pal[1]))





```



```{r figure3dData}
f3d1 <- plot_venn(list("EV" = atcc1_ev, "WCL" = atcc1_w), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8,
                  cat.fontfamily = "sans",
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  ext.percent = c(0,20,0),
                  ext.pos = 45,
                  ext.dist = -0.05,
                  ext.length = 0.5,
                  margin = 0.05)

f3d1 <- as_ggplot(f3d1) + 
  theme(plot.margin = unit(c(1,0.5,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('ATCC10231')", parse = TRUE)
  

f3d2 <- plot_volcano(data = a1_res, 
                     p_val = adj.P.Val, 
                     lfc = logFC,
                     grp = group, 
                     label = ifelse(logFC >= 8, CGD_gene_name, "")) + 
  scale_colour_manual(labels = c("EV enriched (106)", "WCL enriched (134)", "Not significant (445)"),
                      values = unname(ev_pal[c(2,3,1)])) + 
  scale_x_continuous(limits = c(-10,12), expand = c(0,0)) + 
  scale_y_continuous(limits = c(0, 8), expand = c(0,0)) + 
  theme(legend.text = element_text(size = 8)) + 
  annotate("text", x = 8, y = 0.5, label = "bold('ATCC10231')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('106')", parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('134')", parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('445')", parse = TRUE, colour = unname(ev_pal[1]))



```


```{r figure3, fig.cap=f3_cap, fig.width=7.1, fig.height=6, cache=TRUE}

ggarrange(ggarrange(f3a1, f3b1, f3c1, f3d1, 
                    labels = c("A", "", "", ""), 
                    ncol = 4, 
                    nrow = 1, 
                    align = "v"), 
          ggarrange(f3a2, f3b2,
                    f3c2, f3d2, 
                    labels = c("B", rep("", 3)), 
                    ncol = 2, 
                    nrow = 2, 
                    align = "hv", 
                    common.legend = TRUE, 
                    legend = "right", 
                    widths = c(1,1)), 
          ncol = 1,
          nrow = 2, 
          heights = c(1,2))

f3_cap <- "\\textbf{Differential enrichment analysis of proteins identified in \\textit{C. albicans} 
           EVs and whole cell lysates (WCL).} \\textit{(A)} Venn diagrams comparing the EV and WCL 
           proteomes from four \\textit{C. albicans} strains; DAY286 yeast (n = 3), DAY286 biofilm (n = 5), 
           ATCC90028 yeast (n = 3), and ATCC10231 yeast (n = 3). Whole cell lysates were prepared from the EV 
           source cells. Proteins were quantified (MaxLFQ intensity) in minimum 2/3 (yeast) and 4/5 (biofilm) biological 
           replicates of EV or WCL. \\textit{(B)} Volcano plots depicting significantly enriched EV or WCL proteins. 
           Differential enrichment was performed by comparing the mean normalised MaxLFQ intensities of proteins 
           identified in both EV and WCL (i.e. proteins in the venn overlap) using the Bioconductor package 
           \\textit{limma} [@cox2014; @ritchie2015]. Significantly enriched proteins were identified using a 
           Benjamini-Hochberg adjusted p-value cut off of 0.01. Counts of significant and 
           non-significant proteins are indicated on each graph. Proteins with a 
           log\\textsubscript{2}(FC) greater than 8 are labelled. 
           Data tables underlying the venn diagrams and volcano plots are provided in 
           supplemental Files Sx and Sy respectively."
```

\newpage

```{r figure4Data}
# figure 4a GO terms - protein lists
f4a_y_go <- marker_comp[1] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  gsub("CHS3", "CaO19.4937", .) %>% 
  gsub("PEP1", "CaO19.3767", .)

f4a_b_go <- marker_comp[2] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017"), 
                                  vectorize_all = FALSE)

f4a_a1_go <- marker_comp[3] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5", "^MNN2$", 
                                    "MNN22", "MNN24", "ECM331", 
                                    "CAM1-1"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017", "CaO19.2347", 
                                    "CaO19.3803", "CaO19.1995", "CaO19.4255", 
                                    "CaO19.2651"), 
                                  vectorize_all = FALSE)

  
  

f4a_a9_go <- marker_comp[4] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5", "^MNN2$", 
                                    "MNN22", "MNN24", "ECM331", 
                                    "CAM1-1", "FCY21"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017", "CaO19.2347", 
                                    "CaO19.3803", "CaO19.1995", "CaO19.4255", 
                                    "CaO19.2651", "CaO19.8937"), 
                                  vectorize_all = FALSE)
f4a_a9_go <- f4a_a9_go[!f4a_a9_go %in% c("orf19.7397.1", "orf19.7398")]


# figure 4a GO terms bar plots
process_fungifun <- function(filename) {
  df <- read.csv(filename, header = TRUE, sep = "\t", stringsAsFactors = FALSE)
  df$N.prot.found <- as.numeric(gsub(" /.*", "", df$X..genes...category))
  df$N.prot.cat <- as.numeric(gsub(".* / ", "", df$X..genes...category))
  df$N.prot.input <- as.numeric(gsub(".* / ", "", df$X..genes...input))
  df$Cat.ratio <- df$N.prot.found / df$N.prot.cat
  df$Input.ratio <- df$N.prot.found / df$N.prot.input
  
  df <- df[, -c(7:8)]
  
  colnames(df)[1:6] <- c("GO.ID", "GO.term", "GO.type", "Prot.found", "p.val", "adj.p.val")
  
  df$Prot.found <- as.character(df$Prot.found)
  df$Prot.found <- gsub(" | ", ";", df$Prot.found, fixed = TRUE)
  
  return(df)
}

# import FungiFun GO enrichment results
f4a_y_go_table <- process_fungifun("./vignettes/figures/f4a_GO_DAY_Y.csv")
f4a_b_go_table <- process_fungifun("./vignettes/figures/f4a_GO_DAY_B.csv")
f4a_a1_go_table <- process_fungifun("./vignettes/figures/f4a_GO_ATCC10231.csv")
f4a_a9_go_table <- process_fungifun("./vignettes/figures/f4a_GO_ATCC90028.csv")

# Define the colour palette
f4a_palette <- c("#6D2077", "#E4002B", "#FFB81C")

# Bar plots
f4a1 <- plot_go_bar(f4a_y_go_table, c(1:4, 50,53:55, 80,82:84), f4a_palette) + 
  annotate("text", x = 1.5, y = 70, label = "bold('DAY286 yeast')", parse = TRUE) + 
  labs(fill = "GO domain")

f4a2 <- plot_go_bar(f4a_b_go_table, c(1:4, 69,71:73, 96:97, 100:101), f4a_palette) + 
  annotate("text", x = 1.5, y = 65, label = "bold('DAY286 biofilm')", parse = TRUE)

f4a3 <- plot_go_bar(f4a_a1_go_table, c(1:4, 63,65:67, 88:91), f4a_palette) + 
  annotate("text", x = 1.5, y = 75, label = "bold('ATCC10231')", parse = TRUE)

f4a4 <- plot_go_bar(f4a_a9_go_table, c(1:4, 35,37:38,42, 55:56,58:59), f4a_palette) + 
  annotate("text", x = 1.5, y = 75, label = "bold('ATCC90028')", parse = TRUE)



# figure 4b PCA
f4b_data <- cbind(atcc_de, 
                  biofilm_de[match(rownames(atcc_de), rownames(biofilm_de)), ])

f4b_data <- cbind(f4b_data, 
                  yeast_de[match(rownames(f4b_data), rownames(yeast_de)), ])

colnames(f4b_data)[13:28] <- c("B_EV1", "B_EV2", "B_EV3", "B_EV4", "B_EV5", 
                               "B_W1", "B_W2", "B_W3", "B_W4", "B_W5", 
                               "Y_EV1", "Y_EV2", "Y_EV3", 
                               "Y_W1", "Y_W2", "Y_W3")

f4b_labels <- c(rep("EV", 3), rep("WCL", 3), 
                rep("EV", 3), rep("WCL", 3), 
                rep("EV", 5), rep("WCL", 5), 
                rep("EV", 3), rep("WCL", 3))
  
f4b <- plot_mds(f4b_data, mat_labels = f4b_labels, shape_size = 3) + 
  scale_x_continuous(limits = c(-2, 1.5), 
                     expand = c(0,0)) + 
  scale_y_continuous(limits = c(-1.5, 3), 
                     expand = c(0,0)) + 
  scale_fill_manual(breaks = c("A1", "A9", "B", "Y"), 
                    values = unname(rev(st_pal)), 
                    labels = c("A10231", "A90028", "DAY B", "DAY Y"), 
                    guide = "legend") + 
  scale_shape_manual(values = c(22,24)) + 
  theme_bw() + 
  theme(legend.title = element_blank(),
        legend.position = "right", 
        legend.text = element_text(size = 8), 
        legend.direction = "vertical", 
        legend.margin = margin(t = 0, r = 0, b = 0, l = 0), 
        legend.box.margin = margin(t = -10, r = -6, b = -10, l = -10)) + 
  guides(fill = guide_legend(override.aes = list(shape = 21)))
  
# figure 4c 4 way venn
f4c_comp <- list("DAY Y" = rownames(y_ev), 
                 "DAY B" = rownames(b_ev),
                 "ATCC10231" = rownames(atcc1_ev), 
                 "ATCC90028" = rownames(atcc9_ev))
f4c_comp <- lapply(f4c_comp, function(x) match_id(x, ref = uniprot, match = "UP_accession", new = "CGD_gene_name"))

f4c_venn <- plot_venn(vlist = f4c_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 1, 
                      cat.cex = 0.8, 
                      fill = unname(st_pal[c(1,2,4,3)]), 
                      alpha = c(0.3, 0.3, 0.3, 0.3))

f4c <- as_ggplot(f4c_venn) + 
  theme(plot.margin = unit(c(0,2,0,2), "lines"))

f4c_overlap <- plot_venn(vlist = f4c_comp, use_uniprot = FALSE, type = "list") %>% 
  filter(..count.. == 405) %>% 
  pull(..values..) %>% 
  unlist()

# figure 4c GO terms
# marker_comp has list of proteins for each strain
# that are EV enriched or exclusive
# use these for GO enrichment analysis using FungiFun


```


```{r figure4, fig.cap=f4_cap, fig.width=7.1, fig.height=6}

ggarrange(ggarrange(f4a1, f4a2, 
                    f4a3, f4a4, 
                    labels = c("A", rep("", 3)), 
                    ncol = 2, 
                    nrow = 2, 
                    common.legend = TRUE, 
                    legend = "bottom", 
                    align = "hv"), 
          ggarrange(f4b, f4c, 
                    labels = c("B", "C"), 
                    ncol = 2,
                    nrow = 1, 
                    widths = c(1,1)), 
          labels = c("", ""),
          ncol = 1, 
          nrow = 2, 
          heights = c(1.8,1))

f4_cap = "\\textbf{Combined comparison of \\textit{C. albicans} EV proteomes.} \\textit{(A)} 
          Functional enrichment analyses of the significantly enriched and exclusive \\textit{C. albicans} 
          EV proteins. The online tool FungiFun2 (elbe.hki-jena.de/fungifun) 
          was used to identify enriched biological process (BP), 
          cellular component (CC), and molecular function (MF) GO terms [@priebe2015]. A selection of the
          most significantly enriched terms for each GO domain are shown. Full lists of enriched terms can be 
          found in supplemental File Sx. \\textit{(B)} Multidimensional scaling (MDS) plot of each LC-MS/MS
          biological replicate sample shows the sources of variation in the data. Pairwise distance between 
          samples approximately indicates the log\\textsubscript{2}(FC) between samples. Multidimensional scaling 
          was performed using \\texttt{plotMDS} from the Bioconductor package \\textit{limma} [@ritchie2015]. 
          \\textit{(C)} Venn diagram comparing four \\textit{C. albicans} EV proteomes. Lists of proteins 
          present in the centre and right partitions of each venn diagram in Figure \\@ref(fig:figure3)A were 
          compared to identify proteins present in all four datasets. The data table underlying Figure 
          \\@ref(fig:figure4)C is provided in supplemental File Sx."

```

\newpage

```{r figure5Data}
# figure 5c heatmap
clust_dist <- function(x) {
    dist <- cluster::daisy(x, metric = "gower")
    dist[is.na(dist)] <- max(dist, na.rm = TRUE)
    return(dist)
}

f5_table <- a9_res %>% 
  select(logFC, CGD_gene_name) %>% 
  full_join(., select(a1_res, logFC, CGD_gene_name), by = "CGD_gene_name") %>% 
  full_join(., select(yeast_res, logFC, CGD_gene_name), by = "CGD_gene_name") %>% 
  full_join(., select(biofilm_res, logFC, CGD_gene_name), by = "CGD_gene_name") %>% 
  rename(a9_lfc = logFC.x, 
         a1_lfc = logFC.y, 
         y_lfc = logFC.x.x, 
         b_lfc = logFC.y.y) %>% 
  select(CGD_gene_name, everything()) %>% 
  filter(CGD_gene_name %in% f4c_overlap)

# filter for rows with min 1/4 valid values
# otherwise clustering will not work
f5_table_filt <- f5_table %>% 
  filter_at(.vars = vars(contains("lfc")), 
            any_vars(!is.na(.))) %>% 
  tibble::column_to_rownames(var = "CGD_gene_name") %>% 
  as.matrix()

# excluded proteins
f5_table_excl <- f5_table %>% 
  filter_at(.vars = vars(contains("lfc")), 
            all_vars(is.na(.))) %>% 
  tibble::column_to_rownames(var = "CGD_gene_name") %>% 
  as.matrix()

# cluster rows
f5_hr <- hclust(clust_dist(f5_table_filt))

# define row clusters
f5_cutree <- cutree(f5_hr, h = max(f5_hr$height/1.5))

# get cluster ID data for heatmap splitting
f5_split <- f5_cutree %>% 
  as.factor() %>% 
  relevel("5")

# ComplexHeatmap legend colours
col_fun <- circlize::colorRamp2(c(-4,0,4), c("blue", "white", "red"))

# heatmap
f5_htmp <- ComplexHeatmap::Heatmap(f5_table_filt, 
                                    cluster_rows = TRUE,
                                    cluster_columns = TRUE,
                                    clustering_distance_rows = clust_dist, 
                                    clustering_distance_columns = clust_dist, 
                                    na_col = "black", 
                                    col = col_fun, 
                                    show_row_names = FALSE, 
                                    heatmap_legend_param = list(title = expression(bold(log[2]("FC"))), 
                                                                direction = "horizontal", 
                                                                legend_width = unit(2, "cm"), 
                                                                title_position = "lefttop", 
                                                                title_gp = grid::gpar(colour = "black", fontsize = 5), 
                                                                labels_gp = grid::gpar(colour = "black", fontsize = 5), 
                                                                at = c(-6, -4, -2, 0, 2, 4, 6)), 
                                    column_labels = c("ATCC90028", "ATCC10231", "DAY286 Y", "DAY286 B"),
                                    column_names_gp = grid::gpar(colour = "black", fontsize = 6), 
                                    column_names_rot = 45, 
                                    row_title = c("1", "2", "3", "4", "5"),
                                    row_title_gp = grid::gpar(colour = "black", fontsize = 6), 
                                    row_title_rot = 0,
                                    cluster_row_slices = FALSE, 
                                    row_split = f5_split)

# save heatmap as pdf file for editing in inkscape
# not run in rmarkdown but manually in console
# pdf(file = "./vignettes/figures/figure5_unf.pdf", width = 1.7, height = 4)
# ComplexHeatmap::draw(f5_htmp,
#                      heatmap_legend_side = "bottom")
# dev.off()

# protein lists for heatmap table in order of appearance in heatmap
f5_clust1 <- f5_split[f5_hr$order] %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "accession") %>% 
  filter(. == 5) %>% 
  pull(accession)

f5_clust2 <- f5_split[f5_hr$order] %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "accession") %>% 
  filter(. == 1) %>% 
  pull(accession)

f5_clust3 <- f5_split[f5_hr$order] %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "accession") %>% 
  filter(. == 2) %>% 
  pull(accession)

f5_clust4 <- f5_split[f5_hr$order] %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "accession") %>% 
  filter(. == 3) %>% 
  pull(accession)

# cluster 5 has too many proteins to fit in table
# but GO enrichment still possible
f5_clust5 <- f5_split[f5_hr$order] %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "accession") %>% 
  filter(. == 4) %>% 
  pull(accession)

f5_clust6 <- sort(rownames(f5_table_excl))


f5_go1 <- process_fungifun(filename = "./vignettes/figures/f5_GO_clust1.csv")

# issue with PEP1, ambiguous ID in FungiFun, used CaO19.3767 as identifier
f5_go2 <- process_fungifun(filename = "./vignettes/figures/f5_GO_clust2.csv") %>% 
  mutate(Prot.found = gsub("CaO19.3767", "PEP1", Prot.found))


f5_go3 <- process_fungifun(filename = "./vignettes/figures/f5_GO_clust3.csv")

# issue with CHS3, ambiguous ID in FungiFun, used CaO19.4937 as identifier
f5_go4 <- process_fungifun(filename = "./vignettes/figures/f5_GO_clust4.csv") %>% 
  mutate(Prot.found = gsub("CaO19.4937", "CHS3", Prot.found))

# issue with ALA1, MDH1, used CaO19.5746 and CaO19.7481 as identifiers
f5_go5 <- process_fungifun(filename = "./vignettes/figures/f5_GO_clust5.csv") %>% 
  mutate(Prot.found = gsub("CaO19.5746", "ALA1", Prot.found), 
         Prot.found = gsub("CaO19.7481", "MDH1", Prot.found))

f5_go6 <- process_fungifun(filename = "./vignettes/figures/f5_GO_clust6.csv")

```



```{r figure5, fig.cap=f5_cap, fig.height=6}
# load f5
f5 <- png::readPNG("./vignettes/figures/figure5.png")

ggplot() + 
  background_image(f5) + 
  theme(plot.margin = margin(t = 0.1, l = 0.1, r = 0.1, b = 0.1, unit = "inches"))

f5_cap <- "\\textbf{Identification of commonly enriched EV proteins across different \\textit{C. albicans} strains.} 
           Heatmap of proteins (n = 405) identified in all four EV samples. Each column 
           represents a \\textit{C. albicans strain} and each row represents a protein group. The colours
           indicate the log\\textsubscript{2}(FC) of the protein for that particular strain; red is EV enriched, blue is 
           EV depleted, and black is EV exclusive. Unsupervised clustering of rows and columns was performed 
           using Gower's formula via the \\texttt{daisy} function from the R package \\textit{cluster} [@maechler2019].
           Proteins which were exclusive to EVs from all four strains could not be clustered and were 
           seperated prior to clustering, in the group designated \`\`cluster 6\".
           Functional enrichment analyses were performed on the heatmap protein clusters using the online 
           tool FungiFun2 [@priebe2015]. A selection of significantly enriched GO terms are presented in the 
           table inset along with lists of the proteins identified in each cluster."

```


\newpage

```{r figure6Data}

f6a_venn <- plot_venn(vlist = marker_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 0.9, 
                      cat.cex = 0.8, 
                      fill = unname(st_pal[c(1,2,4,3)]), 
                      alpha = c(0.3, 0.3, 0.3, 0.3), 
                      sub = "EV exclusive + sig. enriched", 
                      sub.fontfamily = "sans", 
                      sub.pos = c(0.5, 1.05), 
                      margin = 0.05)

f6a <- as_ggplot(f6a_venn) + 
  theme(plot.margin = unit(c(1.2,0,0,0), "lines"))

wcl_comp <- list("yeast" = yeast_res %>% 
                   filter(group == "WCL up" | group == "WCL ex") %>% 
                   pull(CGD_gene_name), 
                 "biofilm" = biofilm_res %>% 
                   filter(group == "WCL up" | group == "WCL ex") %>% 
                   pull(CGD_gene_name),
                 "ATCC10231" = a1_res %>% 
                   filter(group == "A1_W up" | group == "A1_W ex") %>% 
                   pull(CGD_gene_name), 
                 "ATCC90028" = a9_res %>% 
                   filter(group == "A9_W up" | group == "A9_W ex") %>% 
                   pull(CGD_gene_name))

f6b_venn <- plot_venn(vlist = wcl_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 0.9, 
                      cat.cex = 0.8, 
                      fill = unname(st_pal[c(1,2,4,3)]), 
                      alpha = c(0.3, 0.3, 0.3, 0.3), 
                      sub = "WCL exclusive + sig. enriched", 
                      sub.fontfamily = "sans", 
                      sub.pos = c(0.5, 1.05), 
                      margin = 0.05)

f6b <- as_ggplot(f6b_venn) + 
  theme(plot.margin = unit(c(1.2,0,0,0), "lines"))

```

```{r figure6, fig.cap=f6_cap, fig.width=2.5, fig.height=4}
ggarrange(f6a, f6b, 
          labels = c("A", "B"), 
          ncol = 1,
          nrow = 2)

f6_cap <- "\\textbf{Identification of common signficantly enriched or exclusive \\textit{C. albicans} 
           EV and WCL proteins.} Proteins identified as significantly enriched or exclusive to EVs
           in Figure \\@ref(fig:figure3) were compared to select those which were common to all 
           4 datasets. 43 proteins were identified as potential positive EV protein markers and 60 
           proteins as potential negative EV protein markers. The data tables underlying Figure 
           \\@ref(fig:figure6) are provided in supplemental File Sx. Details of the 47 positive EV
           protein marker candidates and 60 negative candidates can be found in Table  and 
           supplemental Table S2 respectively."
```


\newpage

```{r figure7, fig.cap=f7_cap, fig.width=7.2, fig.height=6}

f7_cap <- "\\textbf{Predicted topology and palmitoylation sites of \\textit{C. albicans} 
           MCC/eisosome proteins SUR7 and orf19.6741 compared to the human tetraspanin CD81.} 
           The presence of transmembrane (TM) domains and palmitoylation sites shown 
           on the left side of the figure were predicted 
           using the online tools TOPCONS2 (topcons.cbr.su.se) [@tsirigos2015] and 
           CSS-PALM 4.0 (csspalm.biocuckoo.org) [@ren2008] respectively.
           For CD81, palmitoylation sites experimentally determined in previous literature 
           are shown [@delandre2009]. The right side of the figure shows a cartoon representation of the 
           TOPCONS2 consensus topology prediction for the three proteins as well as 
           the crystal structure of CD81 (PDB: 5TCX) [@zimmerman2016]."

f7 <- png::readPNG("./vignettes/figures/figure7.png")

ggplot() + 
  background_image(f7)
```

\newpage

```{r figure8, fig.cap=f8_cap, fig.width=3.5, fig.height=2.8}

f8_cap <- "\\textbf{Nineteen best potential \\textit{C. albicans} positive EV protein markers.} 
           These proteins are those underlined in Table . They are 
           are enriched or exclusive to EVs isolated from all of the 
           \\textit{C. albicans} strains investigated in this study. The proteins
           fit within the EV biomarker categories 1a, 2a, and 2b from MISEV2018 which 
           includes transmembrane, GPI-anchored, and cytosolic proteins that are EV 
           enriched [@thery2018]."

f8 <- png::readPNG("./vignettes/figures/figure8.png")

ggplot() + 
  background_image(f8)
```



