---
title: | 
   **Protein markers for *Candida albicans* EVs include claudin-like Sur7 
   family proteins and Rho GTPases**
author: | 
   | Charlotte Dawson^a^, Donovan Garcia Ceron^a^, Harinda Rajapaksha^b^, 
     Pierre Faou^b^,
   | Marilyn Anderson^a^, Mark Bleackley^a^
date: | 
   | *^a^Department of Biochemistry and Genetics, La Trobe Institute for 
     Molecular Science, La Trobe University, 
     Melbourne, 3086, Victoria, Australia*
   | *^b^La Trobe Comprehensive Proteomics Platform, La Trobe Institute for 
     Molecular Science, La Trobe University, Melbourne, 
     3086, Victoria, Australia*
output: 
    bookdown::pdf_document2:
      latex_engine: xelatex
      fig_caption: yes
      number_sections: yes
      toc: no
      keep_tex: no
bibliography: "components/references.bib" 
csl: "components/molecular-and-cellular-proteomics.csl"
link-citations: yes
linkcolor: black
fontsize: 12pt
geometry: margin=1in
classoption: letterpaper
mainfont: Times New Roman
linestretch: 2
header-includes:
  - \usepackage{titling}
  - \pretitle{\begin{center}\fontsize{14bp}{14bp}\selectfont}
  - \posttitle{\par\end{center}}
  - \preauthor{\begin{center}\fontsize{12bp}{12bp}\selectfont}
  - \postauthor{\par\end{center}}
  - \predate{\begin{center}\fontsize{10bp}{10bp}\selectfont}
  - \postdate{\par\end{center}}
  - \usepackage{fancyhdr}
  - \usepackage{lipsum}
  - \pagestyle{fancy}
  - \fancyhead[L]{Defining \textit{Candida albicans} EV protein markers}
  - \setlength{\headheight}{15pt}
  - \usepackage{float}
  - \let\origfigure\figure
  - \let\endorigfigure\endfigure
  - \renewenvironment{figure}[1][2]{\expandafter\origfigure\expandafter[H]}
    {\endorigfigure}
  - \usepackage{caption}
  - \captionsetup{font={normalsize,stretch=1.3},labelfont=bf}
  - \usepackage{threeparttable}
---

**Corresponding author:** Mark Bleackley, m.bleackley@latrobe.edu.au

**Running title:** Defining _Candida albicans_ EV protein markers



\newpage

```{r setup, include=FALSE}
# set knit options
knitr::opts_knit$set(progress = TRUE, 
                     verbose = TRUE)

# set chunk options
knitr::opts_chunk$set(collapse = TRUE,
                      warning = FALSE,
                      message = FALSE,
                      echo = FALSE,
                      comment = "#>")

# load packages
library(candidaev)
library(tidyNano)
library(limma)
library(dplyr)
library(ggplot2)
library(ggpubr)
library(kableExtra)

# define colour palette to distinguish between strains
# colours = yellow, blue, green, pink
# levels = yeast, ATCC90028, ATCC10231, biofilm
st_pal <- c("#E69F00", "#56B4E9", "#009E73", "#CC79A7")

# define colour palette to distinguish between EVs and whole cell lysate (WCL)
# colours = grey, red, blue
# levels = not significant, EV, WCL
ev_pal <- c("#999999", "#D55E00", "#0072B2")
```

```{r yeast, include=FALSE}
# filter out reverse, contaminant proteins, proteins with <2 unique peptides
# extract LFQ intensity columns and UniProt accessions 
# log2 transform LFQ intensities
yeast_lfq <- yeast %>% 
  filter(Reverse != "+", 
         Potential.contaminant != "+", 
         Unique.peptides >= 2) %>% 
  convert_lfq(., yeast_exp)

# drop EV_1 and WCL_1 samples
yeast_lfq2 <- yeast_lfq[, !colnames(yeast_lfq) %in% c("EV_0", "WCL_0")]

# filter for proteins quantified in min 2/3 reps of EV or WCL
yeast_filt2 <- filter_na2(yeast_lfq2, logic = "or", op = "<=", 
                          pat1 = "EV", val1 = 1, 
                          pat2 = "W", val2 = 1)

# normalise LFQ intensities
yeast_norm2 <- normalizeCyclicLoess(yeast_filt2)

# filter for proteins quantified in min 1/3 reps of EV
# define 'EV proteins'
y_ev <- filter_na(yeast_norm2, op = "<=", 
                  pat = "EV", val = 2)

# filter for proteins quantified in min 1/3 reps of WCL
# define 'WCL proteins'
y_wcl <- filter_na(yeast_norm2, op = "<=", 
                   pat = "W", val = 2)

# won't impute with so many missing values
# separate EV/WCL exclusive proteins (don't impute) from 
# EV/WCL common proteins (impute)
yeast_excl <- filter_na2(yeast_norm2, logic = "or", op = "==", 
                         pat1 = "EV", val1 = 3, 
                         pat2 = "W", val2 = 3)

yeast_both <- filter_na2(yeast_norm2, logic = "and", op = "<=", 
                         pat1 = "EV", val1 = 2, 
                         pat2 = "W", val2 = 2)

# use bpca method from pcaMethods for missing value imputation
yeast_imp <- pcaMethods::pca(yeast_both, method = "bpca")
yeast_imp <- pcaMethods::completeObs(yeast_imp)

# recombine imputed proteins and non-imputed proteins in matrix
yeast_de <- rbind(yeast_excl, yeast_imp)

# see limma user guide section 9.2 for more info about DE
# create design matrix
y_samp <- data.frame(T = rep(c("EV", "WCL"), each = 3))

y_design <- stats::model.matrix(~ 0 + T, data = y_samp)
colnames(y_design) <- c("EV", "WCL")

# define sample comparisons of interest
y_contrasts <- c("EV - WCL")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
y_efit <- limma_eBayes(yeast_de, design = y_design, contrasts = y_contrasts)

# extract DE results
# adj.p.val cutoff is 0.01, log2FC cutoff is 1
yeast_res <- get_results(efit = y_efit, 
                         mat = yeast_de, 
                         p_val = 0.01, 
                         lfc = 1, 
                         type = "individual")[[1]]
```

```{r biofilm, include=FALSE}
# filter out reverse, contaminant proteins, proteins with <2 unique peptides
# extract LFQ intensity columns and UniProt accessions 
# log2 transform LFQ intensities
biofilm_lfq <- biofilm %>% 
  filter(Reverse != "+", 
         Potential.contaminant != "+", 
         Unique.peptides >= 2) %>% 
  convert_lfq(., biofilm_exp)

# filter for proteins quantified in min 4/5 reps of EV or WCL
biofilm_filt <- filter_na2(biofilm_lfq, logic = "or", op = "<=", 
                           pat1 = "EV", val1 = 1, 
                           pat2 = "W", val2 = 1)

# normalise LFQ intensities
biofilm_norm <- normalizeCyclicLoess(biofilm_filt)

# filter for proteins quantified in min 1/5 reps of EV
# define 'EV proteins'
b_ev <- filter_na(biofilm_norm, op = "<=", 
                  pat = "EV", val = 4)

# filter for proteins quantified in min 1/5 reps of WCL
# define 'WCL proteins'
b_wcl <- filter_na(biofilm_norm, op = "<=", 
                   pat = "W", val = 4)

# won't impute with so many missing values
# filter for proteins with 4-5 NA values in EV or WCL (don't impute)
biofilm_excl <- filter_na2(biofilm_norm, logic = "or", op = ">=", 
                         pat1 = "EV", val1 = 4, 
                         pat2 = "W", val2 = 4)

# filter for proteins with min 3 valid values in EV and WCL (impute)
biofilm_both <- filter_na2(biofilm_norm, logic = "and", op = "<=", 
                         pat1 = "EV", val1 = 3, 
                         pat2 = "W", val2 = 3)

# use bpca method from pcaMethods for missing value imputation
biofilm_imp <- pcaMethods::pca(biofilm_both, method = "bpca")
biofilm_imp <- pcaMethods::completeObs(biofilm_imp)

# recombine imputed proteins and non-imputed proteins in matrix
biofilm_de <- rbind(biofilm_excl, biofilm_imp)

# see limma user guide section 9.2 for more info about DE
# create design matrix
b_samp <- data.frame(T = rep(c("EV", "WCL"), each = 5))

b_design <- stats::model.matrix(~ 0 + T, data = b_samp)
colnames(b_design) <- c("EV", "WCL")

# define sample comparisons of interest
b_contrasts <- c("EV - WCL")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
b_efit <- limma_eBayes(biofilm_de, design = b_design, contrasts = b_contrasts)

# extract DE results
# adj.p.val cutoff is 0.01, log2FC cutoff is 1
biofilm_res <- get_results(efit = b_efit, 
                           mat = biofilm_de, 
                           p_val = 0.01, 
                           lfc = 1, 
                           type = "individual")[[1]]
```

```{r atccBoth, include=FALSE}
# filter out reverse, contaminant proteins, proteins with <2 unique peptides
# extract LFQ intensity columns and UniProt accessions 
# log2 transform LFQ intensities
atcc_lfq <- atcc %>% 
  filter(Reverse != "+", 
         Potential.contaminant != "+", 
         Unique.peptides >= 2) %>% 
  convert_lfq(., atcc_exp)

# filter for proteins quantified in min 2/3 reps of: 
# A1 EV or A1 WCL or A9 EV or A9 WCL
atcc_filt <- filter_na4(atcc_lfq, "or", "<=", 
                        pat1 = "A10231_EV", 1, 
                        pat2 = "A10231_W", 1, 
                        pat3 = "A90028_EV", 1, 
                        pat4 = "A90028_W", 1)

# normalise LFQ intensities
atcc_norm <- normalizeCyclicLoess(atcc_filt)
```

```{r atcc1, include=FALSE}
# filter for proteins quantified in min 2/3 reps of ATCC10231 EV or WCL
a1_norm <- filter_na2(atcc_norm, logic = "or", op = "<=", 
                      pat1 = "A10231_EV", val1 = 1, 
                      pat2 = "A10231_W", val2 = 1)
a1_norm <- a1_norm[, -c(7:12)]

# filter for proteins quantified in min 1/3 reps of EV
# define 'EV proteins'
a1_ev <- filter_na(a1_norm, op = "<=", pat = "EV", val = 2)

# filter for proteins quantified in min 1/3 reps of WCL
# define 'WCL proteins'
a1_wcl <- filter_na(a1_norm, op = "<=", pat = "W", val = 2)

# won't impute with so many missing values
# separate EV/WCL exclusive proteins (don't impute) from
# EV/WCL common proteins (impute)
a1_excl <- filter_na2(a1_norm, logic = "or", op = "==",
                      pat1 = "EV", val1 = 3,
                      pat2 = "W", val2 = 3)

a1_both <- filter_na2(a1_norm, logic = "and", op = "<=",
                      pat1 = "EV", val1 = 2,
                      pat2 = "W", val2 = 2)

# use bpca method from pcaMethods for missing value imputation
a1_imp <- pcaMethods::pca(a1_both, method = "bpca")
a1_imp <- pcaMethods::completeObs(a1_imp)

# recombine imputed proteins and non-imputed proteins in matrix
a1_de <- rbind(a1_excl, a1_imp)
colnames(a1_de) <- gsub("A10231", "A1", colnames(a1_de))

# see limma user guide section 9.2 for more info about DE
# create design matrix
a1_samp <- data.frame(T = (rep(c("EV", "WCL"), each = 3)))

a1_design <- stats::model.matrix(~ 0 + T, data = a1_samp)
colnames(a1_design) <- c("A1_EV", "A1_W")

# define sample comparisons of interest
a1_contrasts <- c("A1_EV - A1_W")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
a1_efit <- limma_eBayes(a1_de, design = a1_design, contrasts = a1_contrasts)

# extract DE results
# adj.p.val cutoff is 0.01, log2FC cutoff is 1
a1_res <- get_results(efit = a1_efit,
                      mat = a1_de,
                      p_val = 0.01,
                      lfc = 1,
                      type = "individual")[[1]]
```

```{r atcc9, include=FALSE}
# filter for proteins quantified in min 2/3 reps of ATCC90028 EV or WCL
a9_norm <- filter_na2(atcc_norm, logic = "or", op = "<=",
                      pat1 = "A90028_EV", val1 = 1,
                      pat2 = "A90028_W", val2 = 1)
a9_norm <- a9_norm[, -c(1:6)]

# filter for proteins quantified in min 1/3 reps of EV
# define 'EV proteins'
a9_ev <- filter_na(a9_norm, op = "<=", pat = "EV", val = 2)

# filter for proteins quantified in min 1/3 reps of WCL
# define 'WCL proteins'
a9_wcl <- filter_na(a9_norm, op = "<=", pat = "W", val = 2)

# can't impute with so many missing values
# separate EV/WCL exclusive proteins (don't impute) from
# EV/WCL common proteins (impute)
a9_excl <- filter_na2(a9_norm, logic = "or", op = "==",
                      pat1 = "EV", val1 = 3,
                      pat2 = "W", val2 = 3)

a9_both <- filter_na2(a9_norm, logic = "and", op = "<=",
                      pat1 = "EV", val1 = 2,
                      pat2 = "W", val2 = 2)

# use bpca method from pcaMethods for missing value imputation
a9_imp <- pcaMethods::pca(a9_both, method = "bpca")
a9_imp <- pcaMethods::completeObs(a9_imp)

# recombine imputed proteins and non-imputed proteins in matrix
a9_de <- rbind(a9_excl, a9_imp)
colnames(a9_de) <- gsub("A90028", "A9", colnames(a9_de))

# see limma user guide section 9.2 for more info about DE
# create design matrix
a9_samp <- data.frame(T = (rep(c("EV", "WCL"), each = 3)))

a9_design <- stats::model.matrix(~ 0 + T, data = a9_samp)
colnames(a9_design) <- c("A9_EV", "A9_W")

# define sample comparisons of interest
a9_contrasts <- c("A9_EV - A9_W")

# make all pair-wise comparisons between EV and WCL
# and perform limma::eBayes()
a9_efit <- limma_eBayes(a9_de, design = a9_design, contrasts = a9_contrasts)

# extract DE results
# adj.p.val cutoff is 0.01, log2FC cutoff is 1
a9_res <- get_results(efit = a9_efit,
                      mat = a9_de,
                      p_val = 0.01,
                      lfc = 1,
                      type = "individual")[[1]]
```


# Abbreviations

A1 - ATCC10231

A9 - ATCC90028

DAY B - DAY286 biofilm

DAY Y - DAY286 yeast

EV - extracellular vesicle

Evp1 - extracellular vesicle protein 1 (orf19.6741)

Log<sub>2</sub>(FC) - log<sub>2</sub> fold change

MCC - membrane compartment of Can1

MDS - multidimensional scaling

MISEV - minimal information for studies of EVs

sEVs - small EVs

SP - signal peptide

TEMs - tetraspanin enriched microdomains

TM - transmembrane domain

VDM - vesicle depleted medium

WCL - whole cell lysate

\newpage

# Abstract

\newpage

# Introduction

Fungal infections are ubiquitous and are estimated to affect over 20% of the 
world’s population [@havlickova2008]. These infections range from 
non-life-threatening mycoses of the skin and nails, to deadly systemic 
infections. They represent a significant burden to healthcare systems, with the 
direct medical cost of fungal disease in the United States in 2017 estimated to 
be \$7.2 billion [@benedict2018]. The fungi responsible for most fungal 
infections worldwide are the various species of _Candida_ [@pfaller2007]. 
_Candida_ is primarily associated with relatively harmless superficial infections 
[@brown2012], but infections of the bloodstream or internal organs are 
associated with mortality rates as high as 75% [@guinea2014; @kullberg2015]. 
Over 12,000 people were hospitalised for invasive candidiasis in the US in 2017 
alone, at a cost of almost \$1.2 billion [@benedict2018]. The species 
responsible for the majority of _Candida_ infections is _Candida albicans_ 
[@guinea2014]. _C. albicans_ infections are most often treated with two classes 
of antifungals, the azoles which target biosynthesis of the fungal-specific 
sterol ergosterol, and the echinocandins, which target biosynthesis of the major
cell wall polysaccharide 1,3-β-glucan [@pappas2016]. Resistance to these and 
other antifungals has begun to emerge 
[@fisher2018; @perlin2015; @perlin2017; @whaley2017] making the development of 
new antifungals with novel mechanisms of action paramount. Understanding the 
interface between the host and pathogen will help to identify new targets for 
controlling _C. albicans_ infections.

Extracellular vesicles (EVs) are key factors in the progression of many diseases 
ranging from cancer [@xu2018] to neurodegenerative diseases [@quek2017] and 
bacterial infections [@pathirana2016]. EVs are defined as “particles naturally 
released from the cell that are delimited by a lipid bilayer and cannot 
replicate” [@thery2018]. Packaging of cargo into EVs is a method for the 
coordinated release of biological molecules via a non-classic method of 
secretion.  This facilitates intercellular communication both within an 
organism [@maas2017] and between organisms [@schorey2015].

Recently, EVs have 
been proposed to function in the pathogenesis of fungal disease, particularly 
in yeast infections including those from the genera _Cryptococcus_ 
[@bielska2018; @oliveira2010b; @rodrigues2008], _Paracoccidioides_ 
[@dasilva2015a; @vallejo2011], and _Candida_ 
[@joffe2016; @vargas2015; @wolf2015]. The cargo of fungal EVs comprises 
proteins [@bleackley2019], nucleic acids [@dasilva2015], lipids 
[@oliveira2009; @rodrigues2007; @vargas2015; @wolf2015] and carbohydrates 
[@dasilva2015a; @rodrigues2007]. Fungal EVs activate innate immune cells in 
vitro [@dasilva2016; @oliveira2010b; @vargas2015; @wolf2015] and are responsible 
for the hypervirulence of a _Cryptococcus gattii_ strain that caused disease in 
immunocompetent individuals [@bielska2018]. Given the potential roles for EVs in 
host-pathogen interactions, there is significant motivation to better understand 
the biology of fungal EVs.

_C. albicans_ EVs are typical of fungal EVs. They contain proteins, lipids, 
nucleic acids and carbohydrates 
[@dasilva2015; @gil-bona2015b; @vargas2015; @wolf2015; @zarnowski2018]. 
Activation of murine macrophages and dendritic cells by EVs from _C. albicans_ 
indicates a potential role for EVs in modulating the innate immune response to 
the fungus [@vargas2015; @wolf2015]. Formation of biofilms is a critical 
component of _C. albicans_ pathogenesis [@nobile2015] and the biofilm matrix 
provides a protective layer for the fungus against antifungal drugs [@taff2012]. 
The cargo of EVs produced by _C. albicans_ biofilms includes key proteins and 
carbohydrates for biofilm biogenesis, making EVs crucial contributors to the 
establishment of biofilms [@zarnowski2018]. Furthermore, “add-back” of EVs from 
wildtype biofilms restored the loss of azole resistance in biofilms formed by 
strains with deletions in genes encoding important enzymes for matrix 
generation [@zarnowski2018]. That is, EVs also function in antifungal drug 
resistance.

The tools and techniques for the study of fungal EVs have been adapted from 
those established for mammalian systems, where EVs are most intensively 
studied. Many of the cutting-edge experimental techniques for isolation, 
tracking, and analysis of mammalian EVs are dependent on protein markers 
[@lotvall2014]. They include identifying fractions containing EVs after density 
gradient centrifugation [@greening2015] or size exclusion chromatography 
[@gamez-valero2016], purification of EVs via immunoaffinity capture [@oksvold2015], 
and imaging of EV release and uptake [@suetsugu2013]. Unfortunately, proteomic 
analyses of fungal EVs have revealed that the markers that have been so useful 
for the isolation and study of mammalian EVs are not present in fungal EVs 
[@bleackley2019]. Thus, there are differences in the protein cargo and possibly 
the biogenesis of EVs from fungi and mammalian cells, and _de novo_ 
identification of fungal EV markers is essential. Inconsistencies between 
published data sets on the protein content of fungal EVs [@bleackley2019] means 
that a robust proteomic analysis of EVs from a particular species is required 
to provide an adequate data set for EV marker identification. In this study, we 
have addressed this problem by performing in-depth proteomic analyses on 
multiple _C. albicans_ strains to define a set of broadly applicable 
_C. albicans_ EV markers.

We employed label-free quantitative proteomics to generate EV and whole cell 
lysate (WCL) proteomes for a selection of different _C. albicans_ strains and 
morphologies; DAY286 yeast, DAY286 biofilm, ATCC90028 yeast, and ATCC10231 
yeast. By comparing the EV versus WCL data across all four strains we 
identified 39 putative EV markers which are enriched in _C. albicans_ EVs 
regardless of source, cell strain, or morphology. These candidate protein 
markers include GTPases, enzymes crucial for cell wall synthesis, and 
plasma membrane proteins resembling mammalian tetraspanins or claudins. 
The “minimal information for studies of EVs” (MISEV) 2018 position statement 
indicates that, in addition to demonstrating the enrichment of EV marker 
proteins in EV preparations, a deficiency of non-EV proteins should also be 
shown [@thery2018]. With this in mind, we identified 40 proteins that were 
depleted in _C. albicans_ EVs compared to the WCL from the four strains. 
Validation of these putative EV markers in vitro will allow future _C. albicans_ 
EV research to align better with the recommendations set out in MISEV2018. 
Furthermore, robust _C. albicans_ EV markers will allow adaptation of mammalian 
EV analysis techniques for use in the study of EVs in _C. albicans_ pathogenesis 
and will provide potential new targets for the development of novel antifungals. 

# Experimental procedures

## _C. albicans_ strains and maintenance

_C. albicans_ strains used in this study were DAY286 [@davis2002], ATCC90028, 
and ATCC10231. All strains were maintained on YPD Agar plates (1% yeast extract, 
2% peptone, 2% dextrose, 2% agar). Liquid starter cultures were grown overnight 
in YPD medium (1% yeast extract, 2% peptone, 2% dextrose) at 30°C and 300 rpm. 

## Yeast culture conditions

For isolation of EVs from yeast form _C. albicans_ (DAY286, ATCC90028, 
ATCC10231), YPD starter cultures were used to inoculate 150 or 300 mL YPD at an 
OD~600~ = 0.2 in a baffled flask. Cultures were incubated at 30°C overnight 
(approximately 20h). The culture OD~600~ was measured immediately prior to EV 
isolation.

## Biofilm culture conditions

Biofilms (DAY286) were generated based on the method described in [@ramage2002] 
with modifications. Briefly, 500 mL pre-warmed ½ strength RPMI (250 mL RPMI 
1640 (Sigma), 0.165 M MOPS (Sigma) and 250 mL 3.6% dextrose) was inoculated 
using a liquid starter culture to 1.0 x 10^6^ cells/mL. Five 25 x 25 
cm non-treated bioassay dishes (Thermo Fisher Scientific) were each filled with 
100 mL of inoculated ½ RPMI and incubated statically for 4 h at 37°C to allow 
for adhesion of cells to the dish. Following cell adhesion, the culture medium 
and any non-adherent cells was decanted and replaced with 200 mL pre-warmed 
full strength RPMI. Biofilms were formed by incubating the dishes for 54 h at 
37°C with agitation at 28 rpm in an Incu-Shaker Mini (Benchmark Scientific).

## EV isolation

EVs were isolated according to the method described in [@rodrigues2007] with 
minor modifications. In brief, yeast cells or biofilm fragments were separated 
from the culture supernatants by centrifugation at 4,000 x g for 15 min 
(75003607 rotor, Thermo Scientific), then 15,000 x g for 30 min 
(JLA 16.250 rotor, Beckman Coulter). Supernatants were 0.45 μm (Millipore) 
filtered then centrifuged twice at 100,000 x g for 1 h (45Ti rotor, 
Beckman Coulter) to pellet EVs which were resuspended in phosphate buffered 
saline (137 mM NaCl, 2.7 mM KCl, 10 mM Na~2~HPO~4~, 1.8 mM KH~2~PO~4~) (PBS). 
An aliquot of yeast cells (250 µL) was reserved for preparation of yeast whole 
cell lysate. Similarly, 250 μL of biofilm scrapings were resuspended in PBS for 
preparation of biofilm lysate.

## Preparation of whole cell lysate and protein quantification

Samples were washed three times with PBS prior to the addition of approximately 
400 µL of acid washed glass beads (Sigma). Cells were lysed using a TissueLyser 
(Qiagen) at 30 bps for 3 x 1 min bursts with 1 min incubations on ice between 
bursts. Debris and beads were pelleted by centrifugation for 5 min at 
23,000 x g at 4 °C. The supernatant was retained as the respective lysate 
sample. Protein concentration of EVs and whole cell lysates (WCL) was 
determined using a Qubit^TM^ 4 fluorimeter (Life Technologies) and the 
Qubit^TM^ protein assay kit according to the manufacturer’s instructions. 

## Nanoparticle tracking analysis

The particle size and concentration of EV samples was determined using a 
NanoSight NS300 equipped with a 405 nm (blue) laser (Malvern Instruments).  
Immediately prior to injection, EVs were diluted between 1:250 and 1:6000 
in 0.22 μm (Millipore) filtered PBS to bring the particle concentration within 
the manufacturer’s recommended range (10^7^ – 10^9^ particles/mL). NTA software 
version 3.2 was used to record data in 60 s reads in triplicate. Instrument 
settings included: a syringe infusion rate of 70, temperature held at 25°C, 
a camera level of 11, and a detection threshold of 6. The particle 
concentrations and particle size parameters were recorded. Dilution 
factor correction and averaging of data across biological replicates was 
performed using the R package _tidyNano_ [@nguyen2019].

## Transmission electron microscopy

Carbon-coated 400-mesh copper grids (ProSciTech) were glow discharged for 1 
min in a K950X turbo evaporator coupled to a K350 glow discharge unit 
(Quorum Technologies Ltd). EVs (5 µL) were deposited onto the grid and 
incubated for 1 min. Grids were then washed once with ultrapure water and 
negatively stained three times with 2% (v/v) uranyl acetate (Agar Scientific). 
Excess solution was blotted off and the grids were dried overnight. A JEM 2100 
electron microscope (JEOL Ltd) operated at 200 kV was used for imaging.

## Experimental design and statistical rationale

A total of 14 EV and WCL samples from four different _C. albicans_ strains were 
used for proteomics and NTA analyses; DAY286 yeast (n = 3), ATCC90028 yeast 
(n = 3), ATCC10231 yeast (n = 3), and DAY286 biofilm (n = 5). The EV and WCL 
samples from an individual biological replicate were paired as they were derived 
from cells in the same culture or biofilm. Biofilm EVs and WCLs were obtained 
from 5 independent biofilms on different days. Yeast EVs and WCLs were obtained 
from 3 independent yeast cultures in parallel on different days for each strain. 
Preparation of EV and WCL proteomics samples and MS data acquisition was 
performed in three separate batches; DAY286 yeast, DAY286 biofilm, and then 
both ATCC strains. Proteomics samples were randomised prior to injection and 
each biological replicate was injected twice. The raw data from both injections 
was combined into one sample during protein identification and quantification. 
All proteomics statistical and data analyses were performed using R version 
3.6.0 [@rcoreteam2018], Statistical tests, multiple comparison corrections and 
important individual R and Bioconductor packages are referenced where they have 
been used. 

## Sample preparation for mass spectrometry

EVs or WCLs (15 μg protein) were boiled in LDS sample buffer (Life Technologies) 
and TCEP (Thermo Fisher Scientific), then separated by short-range SDS-PAGE 
[@greening2016]. The gel was fixed in 50% (v/v) methanol, 7% (v/v) acetic acid 
for 30 min, and the samples were excised. The proteins were reduced (2 mM TCEP, 
1 h) and alkylated (40 mM iodoacetamide, 30 min in the dark) before digestion 
(1 μg trypsin, 18 h at 37°C). Peptides were then extracted from the gel pieces 
with 85% (v/v) ACN, 0.5% (v/v) TFA, lyophilised and resuspended in 5% (v/v) 
ACN, 0.5% (v/v) TFA.

## Mass spectrometry (ESI-LC-MS/MS) of EVs and WCL proteins

Peptides were analysed as described previously [@fingerhut2018] with 
modifications. Using a nanoflow UPLC instrument (Scientific^TM^ 
UltiMate^TM^ 3000 RSLCnano, Thermo Fisher Scientific), the 
reconstituted peptides were loaded onto a precolumn (C18 PepMap 300 μm ID x 2 cm 
trapping column, Thermo Fisher Scientific) and washed prior to separation using 
a 90 min linear ACN gradient on an analytical column (BEH C18, 1.7 μm, 130 Å 
and 75 μm ID x 25 cm, Waters). The nanoflow UPLC was coupled on-line to a 
Q Exactive HF mass spectrometer (Thermo Fisher Scientific) with a 
nanoelectrospray ion source (Thermo Fisher Scientific). The Orbitrap of the Q 
Exactive HF was used for MS and the linear trap was used for MS/MS.

Data were collected in data-dependent acquisition mode using an MS scan range 
of m/z 350 – 1500 and resolution of 60,000. HCD MS/MS spectra were collected 
for the 15 most intense ions per MS scan at 15,000 resolution with a normalised 
collision energy of 28% and an isolation window of 1.4 m/z. Dynamic exclusion 
parameters were set as follows: exclude isotope on, duration 30 sec and peptide 
match preferred. Other Orbitrap instrument parameters included an MS maximum 
injection time of 30 ms with AGC target 3 x 10^6^, for a maximum 
injection time of 25 ms with an AGT target of 1 x 10^5^. 

## Database search and protein identification

Tandem MS data collected for EVs and WCLs were processed in MaxQuant 
version 1.6.0.16 [@tyanova2016]. Andromeda was used to search the data against 
the UniProt _C. albicans_ reference proteome (UP000000559; downloaded 
16/06/2018; 6035 entries) and the common contaminants and decoys list [@cox2011]. 
Trypsin was selected as the digestion enzyme and up to 2 missed cleavages were 
allowed. Carbamidomethylation of cysteine was set as a fixed modification and 
oxidation of methionine and N-terminal acetylation were set as variable 
modifications. Peptide mass tolerances were 20 ppm (first search) and 4.5 ppm 
(main search), and false discovery rate (FDR) cut offs were 0.01 for both 
proteins and peptides. Minimum and maximum peptide lengths were set to 8 and 25 
respectively and match between runs was not selected. Label-free quantification 
(LFQ) of protein abundances was performed using the MaxLFQ algorithm [@cox2014]. 
Fast LFQ was selected and normalisation was skipped. The proteinGroups.txt 
MaxQuant output was used for further analyses. Metadata for identified proteins 
(localisation, topology, function) were obtained from the UniProt _C. albicans_ 
reference proteome was well as the _C. albicans_ SC5314 Assembly 22 chromosomal 
feature file available from the Candida Genome Database 
(candidagenome.org/download/chromosomal_feature_files) 
[@arnaud2005; @skrzypek2017].

## Protein filtering and proteomics analysis

Data processing and statistical analyses were all performed using R 
version 3.6.0 and RStudio version 1.2.1335 [@rcoreteam2018; @rstudioinc]. 
The DAY286 yeast, DAY286 biofilm, and ATCC proteinGroups.txt files were 
initially processed and analysed separately. Analysis methodology and some 
modified functions were derived from the Bioconductor package _DEP_ [@zhang2018]. 
First, reversed and contaminant proteins as well as proteins identified by less 
than 2 unique peptides were removed. The resulting proteins were filtered for 
those quantified (LFQ intensity > 0) in minimum 2/3 yeast or 4/5 biofilm EV or 
WCL biological replicates. The raw LFQ intensities, sequence coverage, 
number of unique peptides, and MS/MS counts for these proteins can be found in 
supplemental File S1. LFQ intensities were log~2~ transformed, then 
normalised using the default cyclic loess normalisation method from the 
Bioconductor package _limma_ [@ritchie2015]. Proteins were then separated into 
two groups: those exclusive to EVs or WCL, and those quantified in both samples. 
Missing LFQ values in the latter group were imputed using the QRILC method from 
the R package _imputeLCMD_ [@lazar2015], while exclusive proteins were left 
untouched. Finally, differential abundance analysis of proteins in EV versus 
WCL was performed using empirical Bayes method from _limma_ 
[@phipson2016; @ritchie2015]. Protein log~2~(fold changes) indicated 
the difference in average log~2~ LFQ intensity values between EVs and 
WCLs, this value is referred to as the log~2~(FC). The resulting 
p-values were adjusted using the Benjamini-Hochberg approach [@benjamini1995] 
and the significance threshold was set at an adjusted p-value of 0.01. Proteins 
were considered significantly enriched or depleted in EVs if they had a p-value 
less than 0.01 and a positive or negative log~2~(FC) respectively. 
Results of the differential abundance analyses are available in supplemental 
File S2.

## Functional enrichment analysis and clustering

The online tool FungiFun2 version 2.2.8 (sbi.hki-jena.de/fungifun) was used for 
all GO term enrichment analyses [@priebe2015]. Enriched GO terms were those 
with a Benjamini-Hochberg adjusted p-value less than 0.05. Multidimensional 
scaling (MDS) plots were constructed using the plotMDS function from _limma_ 
[@ritchie2015]. The top 500 proteins with the largest standard deviations 
between the proteomics samples were used for the distance calculation. To 
construct the heatmap in Figure 5, unsupervised clustering of log~2~(FC) 
values for the proteins common to EVs from all strains was performed using 
Gower’s formula via the daisy function from the R package _cluster_ [@maechler2019].  

## Protein localisation and attribute prediction

Location information of proteins was obtained from the Candida Genome Database 
(candidagenome.org) [@arnaud2005; @skrzypek2017]. When this information was not 
available DeepLoc-1.0 (cbs.dtu.dk/services/DeepLoc) was used to predict 
subcellular location [@almagroarmenteros2017]. Signal peptide and transmembrane 
domain annotations for each protein were obtained from UniProtKB or were 
predicted using TOPCONS2 (topcons.cbr.su.se) 
[@theuniprotconsortium2019; @tsirigos2015]. Prediction of protein palmitoylation 
sites was performed using CSS-PALM 4.0 (csspalm.biocuckoo.org) [@ren2008].

## _C. albicans_ EV marker definition

Lists of proteins significantly enriched and exclusive in EVs from each strain 
were compared using the R package _VennDiagram_ [@chen2011]. Proteins common to 
all four strains were selected as potential EV positive protein markers. This 
was repeated for significantly enriched and exclusive proteins in WCL to define 
potential EV negative protein markers. Initial candidate EV marker proteins 
were refined using the following criteria to obtain a shortlist of the best 
potential markers. Ideal positive EV markers were proteins that satisfied 
MISEV2018 marker categories 1 and 2 [@thery2018] and had less than 5 transmembrane 
domains, no recognisable signal peptide, and were not detected in 
vesicle-depleted media [@gil-bona2015b]. Ideal negative EV marker proteins were 
selected based on prior tagging with fluorescent proteins in literature, prior 
detection via Western blot, and adherence to MISEV2018 marker categories
3 or 4 [@thery2018].

## Data availability

The proteomics data have been deposited in the ProteomeXchange Consortium 
database via the PRIDE partner repository [@deutsch2017; @perez-riverol2019] 
with the data set identifiers PXD014367 (DAY286 yeast), PXD014388 (ATCC90028 
and ATCC10231), and PXD014389 (DAY286 biofilm). These data include ‘RAW’ files, 
peak list files, MaxQuant search engine txt output, and the UniProt C. albicans 
reference proteome FASTA file. The proteinGroups.txt files and source code used 
in this study is freely available in an online research compendium available at 
github.com/csdaw/candidaev. It was constructed using the R package 
_rrtools_ [@marwick2018] and is structured as an R package which can be used to 
reproduce the results and manuscript as submitted. The compendium has also been 
permanently archived at zenodo (url). 

# Results

# Discussion

# Acknowledgements

# Data availability

# Acknowledgements

\newpage

# References 
\setstretch{1.5}
<div id="refs"></div>


\newpage

# Tables

```{r table1Data, include=FALSE}
# make list of EV sig. enriched and exclusive proteins for each strain
marker_comp <- list("DAY Y" = yeast_res %>% 
                      filter(group == "EV up" | group == "EV ex") %>% 
                      pull(CGD_gene_name), 
                    "DAY B" = biofilm_res %>% 
                      filter(group == "EV up" | group == "EV ex") %>% 
                      pull(CGD_gene_name), 
                    "ATCC90028" = a9_res %>% 
                      filter(group == "A9_EV up" | group == "A9_EV ex") %>% 
                      pull(CGD_gene_name), 
                    "ATCC10231" = a1_res %>% 
                      filter(group == "A1_EV up" | group == "A1_EV ex") %>% 
                      pull(CGD_gene_name))

# get venn segments
marker_comp_tab <- plot_venn(vlist = marker_comp, 
                             use_uniprot = FALSE, 
                             type = "df")

# extract protein names from venn overlap
marker_candidates <- marker_comp_tab %>% 
  filter(apply(.[, 1:4], 1, sum) == 4) %>% 
  pull(..values..) %>% 
  unlist() %>% 
  as.data.frame(stringsAsFactors = FALSE) %>% 
  rename(Name = 1)

# construct table 1
marker_longlist <- marker_candidates %>% 
  mutate(DAY.Y = match_id(Name, yeast_res, "CGD_gene_name", "logFC", 
                          str_split = FALSE), 
         A9 = match_id(Name, a9_res, "CGD_gene_name", "logFC", 
                       str_split = FALSE), 
         A1 = match_id(Name, a1_res, "CGD_gene_name", "logFC", 
                       str_split = FALSE), 
         DAY.B = match_id(Name, biofilm_res, "CGD_gene_name", "logFC", 
                          str_split = FALSE), 
         TM = match_id(Name, uniprot, "CGD_gene_name", "Transmembrane"), 
         SP = match_id(Name, uniprot, "CGD_gene_name", "Signal_peptide")) %>% 
  mutate_at(.vars = vars(DAY.Y, DAY.B, A1, A9), 
            .funs = list(~ sprintf(as.numeric(.), fmt = "%#.2f"))) %>% 
  mutate_at(.vars = vars(DAY.Y, DAY.B, A1, A9), 
            .funs = list(~ case_when(. == "NA" ~ "ex", 
                                     TRUE ~ as.character(.)))) %>% 
  mutate(TM = stringr::str_count(TM, "Helical"), 
         SP = case_when(grepl("SIGNAL", SP) ~ "Y", 
                        TRUE ~ ""), 
         TM = case_when(TM == 0 ~ "", 
                        TRUE ~ as.character(TM))) %>% 
  rename("DAY Y" = DAY.Y, "DAY B" = DAY.B) %>% 
  arrange(Name)

# load .csv file with table 1 labels
marker_labels <- read.csv("components/table1.csv", stringsAsFactors = FALSE) %>% 
  mutate(Function = case_when(Superscript == "a" ~ paste0(Function, footnote_marker_alphabet(1)), 
                              Superscript == "b"~ paste0(Function, footnote_marker_alphabet(2)), 
                              Superscript == "c" ~ paste0(Function, footnote_marker_alphabet(3)), 
                              TRUE ~ as.character(Function)))

# add proper protein functions to table 1
# also add vesicle depleted supernatant info and 
# group proteins according to localisation
marker_longlist2 <- match_id_multi(df = marker_longlist, 
                                   id = "Name", 
                                   ref = marker_labels, 
                                   match = "Name", 
                                   new = c("Function", "Loc_num", "VDM"), 
                                   str_split = FALSE) %>% 
  select(Name, Function, everything()) %>% 
  arrange(Loc_num)

# fix number of TM domains for CDR1;CDR2
marker_longlist2[3, 7] <- "12;12" 

# UniProt says SUR7 has 3 TM domains and a signal peptide
# much of literature says SUR7 has 4 TM domains
# TOPCONS2 predicts SUR7 has 4 TM domains and no signal peptide
# fix SUR7 entries
marker_longlist2[22, 7] <- "4"
marker_longlist2[22, 8] <- ""

# Change name of orf19.6741 to Evp1
marker_longlist2[15, 1] <- paste0("EVP1",
                                  footnote_marker_alphabet(4))
```

<!-- assign markdown references for table caption -->
(ref:table1) [@gil-bona2015]
(ref:footnote1) [@skrzypek2017]
(ref:footnote2) [@theuniprotconsortium2019]
(ref:footnote3) [@tsirigos2015]

```{r table1}
tab1_cap <- "\\textbf{Candidate positive protein markers for 
             \\textit{C. albicans} EVs.} This list of proteins consists of 
             those that were found to be exclusive to EVs or significantly 
             enriched in EVs across the four \\textit{C. albicans} strains 
             examined in this study. Proteins are grouped according to their 
             subcellular localisation as annotated in the Candida Genome 
             Database (candidagenome.org) (ref:footnote1) or in the 
             \\textit{C. albicans} UniProt reference proteome (UP000000559) 
             (ref:footnote2). The log\\textsubscript{2} ratio of the abundance 
             (mean MaxQuant LFQ intensity) of each protein in EVs compared to 
             whole cell lysate (WCL) for each strain is listed. \`\`ex\" 
             indicates where a protein was only quantified in the EV fraction 
             and not the WCL for that strain. The \`\`TM\" column indicates the 
             number of transmembrane domains for each protein as annotated in 
             UniProt. \`\`SP\" indicates whether a protein is annotated as 
             having a signal peptide according to UniProt. \`\`VDM\" shows 
             whether a protein has previously been detected in vesicle-depleted 
             culture media (i.e. the proteins may also be in the soluble secretome) 
             (ref:table1). Underlined proteins are those identified as the 
             best candidates for positive EV markers according to the 
             criteria depicted in supplemental Figure S1."

# draw table 1
knitr::kable(marker_longlist2[ ,-9],
             caption = tab1_cap, 
             format = "latex", 
             booktabs = TRUE, 
             longtable = TRUE, 
             linesep = "", 
             escape = FALSE)  %>% 
  row_spec(row = c(1,2, 6,7, 13,14,15, 18,19,20,21,22,23, 27, 32, 
                   38,39,40, 42,43,44,45), underline = TRUE) %>% 
  add_header_above(c(" " = 2, 
                     "log\\\\textsubscript{2}(fold change) EV vs WCL" = 4, 
                     " " = 2), escape = FALSE) %>% 
  kable_styling(font_size = 8) %>% 
  pack_rows("Plasma membrane", 1, 23) %>% 
  pack_rows("Cell wall, cell surface", 24, 36) %>% 
  pack_rows("Endoplasmic reticulum", 37, 41) %>% 
  pack_rows("Endosome, Golgi, transport vesicle", 42, 44) %>% 
  pack_rows("Vacuole", 45, 45) %>% 
  pack_rows("Mitochondrion", 46, 47) %>% 
  footnote(alphabet = c("Protein localisation was inferred from sequence 
                         similarity with \\\\textit{S. cerevisiae} ortholog as 
                         annotated in the Candida Genome Database 
                        (ref:footnote1).", 
                        "Protein localisation was obtained from the GO Cellular
                        Component annotation in UniProt (ref:footnote2).", 
                        "Presence of transmembrane domains and absence of a 
                        signal peptide was predicted using TOPCONS2 
                        (ref:footnote3).",
                        "The name Evp1 for the protein encoded by orf19.6741 
                        was proposed in the present study."), 
           escape = FALSE, 
           threeparttable = TRUE)
```

\newpage

# Figures

```{r figure1Data, include=FALSE, cache=TRUE}
# construct data.frame with nta experiment info
nta_metadata <- data.frame(id = c("ATCC10231_1", "ATCC10231_2", "ATCC10231_3", 
                                  "ATCC90028_1", "ATCC90028_2", "ATCC90028_3", 
                                  "Biofilm_1", "Biofilm_2", "Biofilm_3", 
                                  "Biofilm_4", "Biofilm_5", 
                                  "DAY286_1", "DAY286_2", "DAY286_3"), 
                           protein_conc = c(0.192, 0.282, 0.287, 
                                            0.424, 0.716, 0.584, 
                                            0.204, 0.198, 0.297, 0.193, 0.416, 
                                            0.437, 0.360, 0.447), 
                           stringsAsFactors = FALSE)

# average the nta technical replicate data
nta_techavg <- nta %>% 
  nanolyze(particle_size, Strain, Bio_rep, Dilution, 
           name = "Tech_rep", 
           param_var = True_count)

# average the nta biological replicate data
nta_bioavg <- nta_techavg %>% 
  nanolyze(particle_size, Strain,
           name = "Bio_rep", 
           param_var = Tech_rep_mean) %>% 
  mutate(facet = case_when(Strain == "Biofilm" ~ "second", TRUE ~ "first"))

# determine the total particle concentration for each biological replicate
nta_total <- nta_techavg %>% 
  nanocount(Strain, Bio_rep, Dilution, 
            param_var = Tech_rep_mean) %>% 
  tidyr::unite("id", Strain, Bio_rep) %>% 
  mutate(protein_conc = as.numeric(match_id(id, nta_metadata, 
                                            "id", "protein_conc")), 
         od = match_id(id, nta_metadata, "id", "OD600"), 
         part_per_prot = Particle_count / protein_conc / 1000) %>% 
  tidyr::separate(col = id, into = c("Strain", "Bio_rep"), sep = "_", 
                  remove = TRUE)

# use ANOVA with Tukey HSD to compare mean particle per ug protein 
# ratio across the strains
nta_total_tukey <- aov(part_per_prot ~ Strain, data = nta_total) %>% 
  tukey_hsd()
```


```{r figure1plots, include=FALSE}
# construct figure 1a
f1a_img <- png::readPNG("figures/figure1a.png")

f1a <- ggplot() + 
  background_image(f1a_img) + 
  theme(plot.margin = margin(t = 0.3, l = 0, r = 0, b = 0, unit = "inches"))

# construct figure 1b
f1b <- nta_bioavg %>% 
  ggplot(aes(x = particle_size, y = Bio_rep_mean, colour = Strain)) + 
  scale_colour_manual(values = st_pal[c(3, 2, 4, 1)], 
                      breaks = c("DAY286", "ATCC90028", 
                                 "ATCC10231", "Biofilm"), 
                      labels = c("DAY286 Y", "ATCC90028", 
                                 "ATCC10231", "DAY286 B")) + 
  geom_vline(xintercept = 100, colour = "#999999", linetype = "dashed") +
  geom_line(size = 0.5) + 
  facet_wrap(~ facet, scales = "free", ncol = 2, nrow = 1) + 
  theme_classic() + 
  theme(strip.background = element_blank(), 
        strip.text = element_blank(), 
        legend.position = c(0.88, 0.6), 
        legend.title = element_blank(), 
        plot.margin = margin(t = 0.2, l = 0, r = 0.2, b = 0, unit = "inches"), 
        axis.text.x = element_text(colour = "black")) + 
  xlab(label = "Particle size (nm)") + 
  ylab(label = "Particles / mL")

# construct figure 2c
nta_total$Strain <- factor(nta_total$Strain, 
                           levels = c("DAY286", "ATCC90028", 
                                      "ATCC10231", "Biofilm"))

f1c <- nta_total %>% 
  ggboxplot(x = "Strain", y = "part_per_prot", 
            color = "Strain", palette = st_pal, add = "jitter", 
            add.params = list(color = "black"), 
            shape = 15,
            ylab = expression("Particles /"~paste(mu,g)~"protein"))
f1c <- ggpar(f1c, 
             yscale = "log10", 
             format.scale = TRUE, 
             ylim = c(8e8, 8e10),
             font.y = 10, 
             legend = "none") + 
  stat_pvalue_manual(nta_total_tukey[c(2,4,6), ], label = "p.adj", 
                     y.position = c(10.45, 10.65, 10.85), 
                     tip.length = 1e-2, 
                     label.size = 3) + 
  theme(axis.title.x = element_blank(), 
        plot.margin = margin(t = 0.2, l = 0.2, r = 0, b = 0.15, 
                             unit = "inches")) + 
  scale_x_discrete(labels = c("ATCC90028" = "A9", 
                              "ATCC10231" = "A1", 
                              "Biofilm" = "B", 
                              "DAY286" = "Y"))
```


```{r figure1, fig.cap=f1_cap, fig.width=7.1, fig.height=4}
# draw figure 1
ggarrange(f1a, 
          ggarrange(f1b, f1c, 
                    labels = c("B", "C"), 
                    ncol = 2,
                    nrow = 1, 
                    widths = c(1.8,1)), 
          labels = c("A", ""), 
          nrow = 2, 
          ncol = 1, 
          heights = c(1,1))

f1_cap <- "\\textbf{Characterisation and quantification of EVs from different 
           \\textit{C. albicans} strains.} \\textit{(A)} Representative TEM 
           images of EVs isolated from 4 different \\textit{C. albicans} 
           strains. Scale bar indicates 0.5 $\\mu$m. \\textit{(B)} Size 
           distribution of DAY286 yeast (n = 3), ATCC90028 yeast (n = 3), 
           ATCC10231 yeast (n = 3), and DAY286 biofilm (n = 5) EVs as measured 
           by nanoparticle tracking analysis. Line plots shown are the average 
           of the biological replicates and the dashed grey line indicates 100 
           nm. Plots for the individual biological replicates are presented in 
           supplemental Figure S2. \\textit{(C)} Comparison of the ratio of 
           particle concentration to protein concentration of the EV fraction, 
           (particles/mL) $\\div$ ($\\mu$g protein/mL), across the four EV 
           sources. Each dot represents one biological replicate. Sample means 
           were compared using ANOVA followed by Tukey's HSD \\textit{post-hoc} 
           test. P-values indicating significant differences are shown. The 
           particle and protein concentrations for each biological replicate 
           are provided in supplemental Table S1."
```

\newpage

```{r figure2Yeast, include=FALSE}
# compare DAY286 yeast EV WCL proteins
f2a1 <- plot_venn(list("EV" = y_ev, "WCL" = y_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8,
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  margin = 0.05)

# DAY286 yeast venn
f2a1 <- as_ggplot(f2a1) + 
  theme(plot.margin = unit(c(1,0.5,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, 
           label = "bold('DAY286 yeast')", parse = TRUE)

# generate volcano plot
f2b1 <- plot_volcano(data = yeast_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     use_labels = TRUE,
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched", 
                                 "WCL enriched", 
                                 "Not significant"), 
                      values = unname(ev_pal[c(2,1,3)]), 
                      breaks = c("EV up", "WCL up", "not sig"), 
                      name = element_blank()) + 
  annotate("text", x = 8.5, y = 0.5, label = "bold('DAY286 Y')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('101')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('117')", parse = TRUE, 
           colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('276')", parse = TRUE, 
           colour = unname(ev_pal[1]))
```



```{r figure2A9, include=FALSE}
# compare ATCC90028 EV WCL proteins
f2a2 <- plot_venn(list("EV" = a9_ev, "WCL" = a9_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans",
                  cex = 0.8,
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.55)), 
                  ext.percent = c(30,10,0),
                  ext.pos = c(45,315),
                  ext.dist = -0.05,
                  ext.length = 0.5,
                  margin = 0.05)

# ATCC90028 venn
f2a2 <- as_ggplot(f2a2) + 
  theme(plot.margin = unit(c(1,0,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('ATCC90028')", parse = TRUE)

# generate volcano plot
f2b2 <- plot_volcano(data = a9_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     use_labels = TRUE, 
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched", "WCL enriched", 
                                 "Not significant"),
                      values = unname(ev_pal[c(2,3,1)])) + 
  annotate("text", x = 8, y = 0.5, label = "bold('ATCC90028')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('121')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('176')", 
           parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('672')", 
           parse = TRUE, colour = unname(ev_pal[1]))
```



```{r figure2A1, include=FALSE}
# compare ATCC10231 EV WCL proteins
f2a3 <- plot_venn(list("EV" = a1_ev, "WCL" = a1_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8,
                  cat.fontfamily = "sans",
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  ext.percent = c(0,20,0),
                  ext.pos = 45,
                  ext.dist = -0.05,
                  ext.length = 0.5,
                  margin = 0.05)

# ATCC10231 venn
f2a3 <- as_ggplot(f2a3) + 
  theme(plot.margin = unit(c(1,0.5,1,0.5), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('ATCC10231')", parse = TRUE)
  
# generate volcano plot
f2b3 <- plot_volcano(data = a1_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     use_labels = TRUE, 
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched", "WCL enriched", 
                                 "Not significant"),
                      values = unname(ev_pal[c(2,3,1)])) + 
  annotate("text", x = 8, y = 0.5, label = "bold('ATCC10231')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('101')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('105')", 
           parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('468')", 
           parse = TRUE, colour = unname(ev_pal[1]))
```



```{r figure2Biofilm, include=FALSE}
# compare DAY286 biofilm EV WCL proteins
f2a4 <- plot_venn(list("EV" = b_ev, "WCL" = b_wcl), 
                  use_uniprot = TRUE, type = "plot", 
                  fontfamily = "sans", 
                  cex = 0.8, 
                  cat.fontfamily = "sans", 
                  cat.just = list(c(0.5,1.7), c(0.5,1.5)), 
                  ext.text = TRUE,
                  ext.percent = c(20,10,0),
                  ext.pos = c(45,315),
                  ext.dist = -0.1,
                  ext.length = 0.5,
                  margin = 0.05)
# DAY286 biofilm venn
f2a4 <- as_ggplot(f2a4) + 
  theme(plot.margin = unit(c(1,0,1,0), "lines")) + 
  annotate("text", x = 0.5, y = 0.92, label = "bold('DAY286 biofilm')", 
           parse = TRUE)

# generate volcano plot
f2b4 <- plot_volcano(data = biofilm_res, 
                     p_val = adj.P.Val, 
                     log2fc = logFC,
                     group = group, 
                     use_labels = TRUE, 
                     label_fun = ifelse(logFC >= 8, CGD_gene_name, ""), 
                     point_size = 0.6, 
                     label_size = 2, 
                     x_lim = c(-10, 12), 
                     y_lim = c(0, 8), 
                     legend_params = c("left", 11, 8), 
                     axis_params = c("black", 10)) + 
  scale_colour_manual(labels = c("EV enriched", "WCL enriched", 
                                 "Not significant"), 
                      values = unname(ev_pal[c(2,1,3)]), 
                      breaks = c("EV up", "WCL up", "not sig")) + 
  annotate("text", x = 8.5, y = 0.5, label = "bold('DAY286 B')", parse = TRUE) + 
  annotate("text", x = 3.5, y = 7, label = "bold('105')", 
           parse = TRUE, colour = unname(ev_pal[2])) + 
  annotate("text", x = -3.5, y = 7, label = "bold('100')", 
           parse = TRUE, colour = unname(ev_pal[3])) + 
  annotate("text", x = 0, y = 6, label = "bold('453')", 
           parse = TRUE, colour = unname(ev_pal[1]))
```



```{r figure2, fig.cap=f2_cap, fig.width=7.1, fig.height=6}
# draw figure 2
ggarrange(ggarrange(f2a1, f2a2, f2a3, f2a4, 
                    labels = c("A", "", "", ""), 
                    ncol = 4, 
                    nrow = 1, 
                    align = "v"), 
          ggarrange(f2b1, f2b2,
                    f2b3, f2b4, 
                    labels = c("B", rep("", 3)), 
                    ncol = 2, 
                    nrow = 2, 
                    align = "hv", 
                    common.legend = TRUE, 
                    legend = "right", 
                    widths = c(1,1), 
                    vjust = 1), 
          ncol = 1,
          nrow = 2, 
          heights = c(1,2))

f2_cap <- "\\textbf{Differential abundance analysis of proteins identified in 
           \\textit{C. albicans} EVs and whole cell lysates (WCL).} 
           \\textit{(A)} Venn diagrams comparing the EV and WCL proteomes from 
           four \\textit{C. albicans} strains; DAY286 yeast (n = 3), ATCC90028 
           yeast (n = 3), ATCC10231 yeast (n = 3), and DAY286 biofilm (n = 5). 
           Whole cell lysates were prepared from the EV source cells. Proteins 
           were quantified (MaxQuant LFQ intensity) in minimum 2/3 (yeast) and 
           4/5 (biofilm) biological replicates of EV or WCL. \\textit{(B)} 
           Volcano plots depicting significantly enriched EV or WCL proteins. 
           Differential abundance analysis was performed by comparing the mean 
           normalised LFQ intensities of proteins identified in both EV and 
           WCL (i.e. proteins in the venn overlap) using the Bioconductor 
           package \\textit{limma} [@cox2014; @ritchie2015]. Significantly 
           enriched proteins were identified using a Benjamini-Hochberg 
           adjusted p-value cut off of 0.01 and log\\textsubscript{2}(FC) cutoff 
           of 1. Counts of significant and 
           non-significant proteins are indicated on each graph. Proteins with 
           a log\\textsubscript{2}(FC) greater than 8 are labelled. Data tables 
           underlying the venn diagrams and volcano plots are provided in 
           supplemental File S2."
```

\newpage

```{r figure3Data, include=FALSE}
# get list of DAY286 yeast EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f3_y_go <- marker_comp[1] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  gsub("CHS3", "CaO19.4937", .) %>% 
  gsub("PEP1", "CaO19.3767", .)
# paste into FungiFun2 and download results .csv

# get list of DAY286 biofilm EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f3_b_go <- marker_comp[2] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017"), 
                                  vectorize_all = FALSE)
# paste into FungiFun2 and download results .csv

# get list of ATCC90028 EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f3_a9_go <- marker_comp[3] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5", "^MNN2$", 
                                    "MNN22", "MNN24", "ECM331", 
                                    "CAM1-1", "FCY21"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017", "CaO19.2347", 
                                    "CaO19.3803", "CaO19.1995", "CaO19.4255", 
                                    "CaO19.2651", "CaO19.8937"), 
                                  vectorize_all = FALSE)
f3_a9_go <- f3_a9_go[!f3_a9_go %in% c("orf19.7397.1", "orf19.7398")]
# paste into FungiFun2 and download results .csv

# get list of ATCC10231 EV proteins exclusive and sig. enriched
# substitute protein names that don't work with FungiFun2
f3_a1_go <- marker_comp[4] %>% 
  unlist() %>% 
  strsplit(., ";") %>% 
  unlist() %>% 
  stringi::stri_replace_all_regex(., 
                                  c("CHS3", "PEP1", "ALS3", 
                                    "PLB3", "PLB4.5", "^MNN2$", 
                                    "MNN22", "MNN24", "ECM331", 
                                    "CAM1-1"), 
                                  c("CaO19.4937", "CaO19.3767", "CaO19.1816",
                                    "CaO19.6594", "CaO19.9017", "CaO19.2347", 
                                    "CaO19.3803", "CaO19.1995", "CaO19.4255", 
                                    "CaO19.2651"), 
                                  vectorize_all = FALSE)
# paste into FungiFun2 and download results .csv
```

```{r figure3Plots, include=FALSE}
# import FungiFun2 GO enrichment results
f3_y_go_table <- process_fungifun("figures/f3_GO_DAY_Y.csv")
f3_b_go_table <- process_fungifun("figures/f3_GO_DAY_B.csv")
f3_a1_go_table <- process_fungifun("figures/f3_GO_ATCC10231.csv")
f3_a9_go_table <- process_fungifun("figures/f3_GO_ATCC90028.csv")

# define the colour palette
f3_pal <- c("#6D2077", "#E4002B", "#FFB81C")

# GO term bar plots
f3_y <- plot_go_bar(f3_y_go_table, c(1:4, 49,52:53,55, 77,79:81), f3_pal) + 
  annotate("text", x = 12, y = 65, label = "bold('DAY286 Y')", parse = TRUE) + 
  labs(fill = "GO domain") + 
  theme(legend.text = element_text(size = 10), 
        legend.title = element_text(size = 10))

f3_b <- plot_go_bar(f3_b_go_table, c(1:4, 69,71:73, 96:97, 100, 102), f3_pal) + 
  annotate("text", x = 12, y = 60, label = "bold('DAY286 B')", parse = TRUE)

f3_a9 <- plot_go_bar(f3_a9_go_table, c(1:3,6, 27,29,33,40, 46,49,51:52), f3_pal) + 
  annotate("text", x = 12, y = 70, label = "bold('ATCC90028')", parse = TRUE)

f3_a1 <- plot_go_bar(f3_a1_go_table, c(1:4, 62,65:66,76, 86:89), f3_pal) + 
  annotate("text", x = 12, y = 70, label = "bold('ATCC10231')", parse = TRUE)
```



```{r figure3, fig.cap=f3_cap, fig.width=7.2, fig.height=5}
# draw figure 4
ggarrange(f3_y, f3_a9, 
          f3_a1, f3_b, 
          labels = NULL, 
          ncol = 2, 
          nrow = 2, 
          common.legend = TRUE, 
          legend = "bottom", 
          align = "hv")

f3_cap <- "\\textbf{Functional enrichment analyses of the significantly 
           enriched and exclusive \\textit{C. albicans} EV proteins.} The 
           online tool FungiFun2 (elbe.hki-jena.de/fungifun) was used to 
           identify enriched biological process (BP), cellular component (CC), 
           and molecular function (MF) GO terms [@priebe2015]. A selection of 
           the most significantly enriched terms for each GO domain are shown. 
           They are presented top to bottom in order of increasing 
           Benjamini-Hochberg adjusted p-value. Full lists of enriched GO 
           terms can be found in supplemental File S3."
```

\newpage

```{r figure4Data, include=FALSE}
# figure 4a EV venn
f4a_comp <- list("DAY Y" = rownames(y_ev), 
                 "DAY B" = rownames(b_ev),
                 "ATCC90028" = rownames(a9_ev), 
                 "ATCC10231" = rownames(a1_ev))
f4a_comp <- lapply(f4a_comp, function(x) match_id(x, ref = uniprot, 
                                                  match = "UP_accession", 
                                                  new = "CGD_gene_name"))

f4a_venn <- plot_venn(vlist = f4a_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 1, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3))

f4a <- as_ggplot(f4a_venn) + 
  theme(plot.margin = unit(c(0,2,0,2), "lines"))

f4a_overlap <- plot_venn(vlist = f4a_comp, 
                         use_uniprot = FALSE, type = "df") %>% 
  filter(..count.. == 396) %>% 
  pull(..values..) %>% 
  unlist()

# figure 4b WCL venn
f4b_comp <- list("DAY Y" = rownames(y_wcl), 
                 "DAY B" = rownames(b_wcl),
                 "ATCC90028" = rownames(a9_wcl), 
                 "ATCC10231" = rownames(a1_wcl))
f4b_comp <- lapply(f4b_comp, function(x) match_id(x, ref = uniprot, 
                                                  match = "UP_accession", 
                                                  new = "CGD_gene_name"))

f4b_venn <- plot_venn(vlist = f4b_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 1, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3))

f4b <- as_ggplot(f4b_venn) + 
  theme(plot.margin = unit(c(0,2,0,2), "lines"))

# MDS data table
mds_a9 <- a9_de %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "id")

mds_a1 <- a1_de %>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "id")

mds_biofilm <- biofilm_de%>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "id")

mds_yeast <- yeast_de%>% 
  as.data.frame() %>% 
  tibble::rownames_to_column(var = "id")

mds_data <- full_join(mds_a9, mds_a1, by = "id") %>% 
  full_join(., mds_biofilm, by = "id") %>% 
  full_join(., mds_yeast, by = "id") %>% 
  tibble::column_to_rownames(var = "id")

colnames(mds_data)[13:28] <- c("B_EV1", "B_EV2", "B_EV3", "B_EV4", "B_EV5", 
                               "B_W1", "B_W2", "B_W3", "B_W4", "B_W5", 
                               "Y_EV1", "Y_EV2", "Y_EV3", 
                               "Y_W1", "Y_W2", "Y_W3")

mds_labels <- c(rep("EV", 3), rep("WCL", 3), 
                rep("EV", 3), rep("WCL", 3), 
                rep("EV", 5), rep("WCL", 5), 
                rep("EV", 3), rep("WCL", 3))

# figure 4c MDS [dim1, dim2]
f4c <- plot_mds(mds_data, mat_labels = mds_labels, 
                shape_size = 3, dim.plot = c(1,2)) + 
  scale_x_continuous(limits = c(-1.5, 2), 
                     expand = c(0,0)) + 
  scale_y_continuous(limits = c(-2, 1.5), 
                     expand = c(0,0)) + 
  scale_fill_manual(breaks = c("Y", "A9", "A1", "B"), 
                    values = st_pal[c(3, 2, 4, 1)], 
                    labels = c("DAY Y", "A90028", "A10231", "DAY B"), 
                    guide = "legend") + 
  scale_shape_manual(values = c(22,24)) + 
  theme_bw() + 
  theme(legend.title = element_blank(),
        legend.position = "right", 
        legend.text = element_text(size = 8), 
        legend.direction = "vertical") + 
  guides(fill = guide_legend(override.aes = list(shape = 21))) + 
  xlab(label = "Dimension 1") + 
  ylab(label = "Dimension 2")
  
# figure 4d MDS [dim3, dim2]
f4d <- plot_mds(mds_data, mat_labels = mds_labels, 
                shape_size = 3, dim.plot = c(3,2)) + 
  scale_x_continuous(limits = c(-1.5, 2), 
                     expand = c(0,0)) + 
  scale_y_continuous(limits = c(-2, 1.5), 
                     expand = c(0,0)) + 
  scale_fill_manual(breaks = c("Y", "A9", "A1", "B"), 
                    values = st_pal[c(3, 2, 4, 1)], 
                    labels = c("DAY Y", "A90028", "A10231", "DAY B"), 
                    guide = "legend") + 
  scale_shape_manual(values = c(22,24)) + 
  theme_bw() + 
  theme(legend.title = element_blank(),
        legend.position = "right", 
        legend.text = element_text(size = 8), 
        legend.direction = "vertical", 
        legend.margin = margin(t = 0, r = 0, b = 0, l = 0.1), 
        legend.box.margin = margin(t = -10, r = -6, b = -10, l = -10)) + 
  guides(fill = guide_legend(override.aes = list(shape = 21))) + 
  xlab(label = "Dimension 3") + 
  ylab(label = "Dimension 2")
```

```{r figure4, fig.cap=f4_cap, fig.width=7.1, fig.height=5}
ggarrange(ggarrange(f4a, f4b,
                    labels = c("A", "B"),
                    ncol = 2,
                    nrow = 1),
          ggarrange(f4c, f4d,
                    labels = c("C", "D"),
                    ncol = 2,
                    nrow = 1, 
                    common.legend = TRUE,
                    legend = "right", 
                    align = "hv"),
          labels = NULL,
          ncol = 1,
          nrow = 2,
          heights = c(2, 3))

f4_cap <- "\\textbf{Combined comparison of \\textit{C. albicans} proteomes.} 
           \\textit{(A)} Venn diagram comparing four \\textit{C. albicans} EV 
           proteomes. Lists of proteins present in the centre and right 
           partitions of each venn diagram in Figure \\@ref(fig:figure2)A were 
           compared to identify proteins present in all four datasets. 
           \\textit{(B)} Venn diagram comparing four \\textit{C. albicans} WCL 
           proteomes. Lists of proteins present in the centre and left 
           partitions of each venn diagram in Figure \\@ref(fig:figure2)A were 
           compared to identify proteins present in all four datasets. The data 
           tables underlying Figures \\@ref(fig:figure4)A and 
           \\@ref(fig:figure4)B are provided in supplemental File S4. 
           \\textit{(C)} Multidimensional scaling (MDS) plot of each LC-MS/MS 
           biological replicate sample with the first 2 dimensions shown. 
           Pairwise distance between samples approximately indicates the 
           log\\textsubscript{2}(FC) between samples. \\textit{limma} was used 
           to generate the MDS plots [@ritchie2015]. \\textit{(D)} MDS plot of 
           each LC-MS/MS biological replicate sample with the second and third 
           dimensions shown."
```

\newpage

```{r figure5Heatmap, include=FALSE}
# figure 5 heatmap
f5_table <- a9_res %>% 
  select(logFC, CGD_gene_name) %>% 
  full_join(., select(a1_res, logFC, CGD_gene_name), 
            by = "CGD_gene_name") %>% 
  full_join(., select(yeast_res, logFC, CGD_gene_name), 
            by = "CGD_gene_name") %>% 
  full_join(., select(biofilm_res, logFC, CGD_gene_name), 
            by = "CGD_gene_name") %>% 
  rename(a9_lfc = logFC.x, 
         a1_lfc = logFC.y, 
         y_lfc = logFC.x.x, 
         b_lfc = logFC.y.y) %>% 
  select(CGD_gene_name, everything()) %>% 
  filter(CGD_gene_name %in% f4a_overlap)

# filter for rows with min 1/4 valid values
# otherwise clustering will not work
f5_table_filt <- f5_table %>% 
  filter_at(.vars = vars(contains("lfc")), 
            any_vars(!is.na(.))) %>% 
  tibble::column_to_rownames(var = "CGD_gene_name") %>% 
  as.matrix()

# excluded proteins
f5_table_excl <- f5_table %>% 
  filter_at(.vars = vars(contains("lfc")), 
            all_vars(is.na(.))) %>% 
  tibble::column_to_rownames(var = "CGD_gene_name") %>% 
  as.matrix()

# heatmap legend parameters
f5_params <- list(color_bar = "continuous", 
                  title = expression(bold(log[2]("FC"))), 
                  direction = "horizontal", 
                  legend_width = unit(2, "cm"), 
                  title_position = "lefttop", 
                  title_gp = grid::gpar(colour = "black", 
                                        fontsize = 5), 
                  labels_gp = grid::gpar(colour = "black", 
                                         fontsize = 5), 
                  at = seq(-6, 6, 2))

# plot heatmap
f5_htmp <- plot_heatmap(mt = f5_table_filt, 
                        plot = FALSE,
                        df = FALSE, 
                        data_type = "log2fc", 
                        clust_fun = "gower", 
                        split_type = "cutree", 
                        k = 5, 
                        cluster_split = FALSE, 
                        split_order = c(2, 5, 1, 4, 3), 
                        colour_lims = c(-4, 4),
                        column_labels = c("ATCC90028", "ATCC10231", 
                                          "DAY286 Y", "DAY286 B"), 
                        col_name_fontsize = 6, 
                        column_names_rot = 45, 
                        row_title_fontsize = 6, 
                        row_title_rot = 0, 
                        row_title = c("1", "2", "3", "4", "5"), 
                        heatmap_legend_param = f5_params)

# save heatmap as pdf file for editing in inkscape
# not run in rmarkdown but manually in console
# pdf(file = "inst/manuscript/figures/figure5_htmp.pdf", width = 1.7, height = 4)
# ComplexHeatmap::draw(f5_htmp, heatmap_legend_side = "bottom")
# dev.off()
```

```{r figure5Table, include=FALSE}
#  list of proteins for heatmap table in order of appearance in heatmap
f5_clusters <- plot_heatmap(mt = f5_table_filt, 
                            plot = FALSE,
                            df = TRUE, 
                            data_type = "log2fc", 
                            clust_fun = "gower", 
                            split_type = "cutree", 
                            k = 5, 
                            cluster_split = FALSE) %>% 
  mutate(cluster = recode(cluster, 
                          "1" = 4, "2" = 1, "3" = 5, "4" = 2, "5" = 3)) %>% 
  arrange(cluster)

# extract identifiers from each cluster
f5_clust1 <- f5_clusters %>% 
  filter(cluster == 1) %>% 
  mutate(id = gsub("PEP1", "CaO19.3767", id)) %>% 
  pull(id) %>% 
  stringr::str_extract_all(., "[^;]+") %>% 
  unlist()

f5_clust2 <- f5_clusters %>% 
  filter(cluster == 2) %>% 
  pull(id)
  
f5_clust3 <- f5_clusters %>% 
  filter(cluster == 3) %>% 
  mutate(id = gsub("CHS3", "CaO19.4937", id)) %>% 
  pull(id)

f5_clust4 <- f5_clusters %>% 
  filter(cluster == 4) %>% 
  pull(id)

# cluster 5 has too many proteins to fit in table
# but GO enrichment still possible
f5_clust5 <- f5_clusters %>% 
  filter(cluster == 5) %>% 
  mutate(id = gsub("ALA1", "CaO19.5746", id), 
         id = gsub("MDH1", "CaO19.7481", id), 
         id = gsub("RPS1", "CaO19.3002", id), 
         id = gsub("TSA1", "CaO19.7417", id)) %>% 
  pull(id) %>% 
  stringr::str_extract_all(., "[^;]+") %>% 
  unlist() %>% 
  unique()

# cluster 6 is proteins that could not be clustered
# because all 4 log2FCs = NA
f5_clust6 <- sort(rownames(f5_table_excl))

# import FungiFun2 results
f5_go1 <- process_fungifun("figures/f5_GO_clust1.csv") %>% 
  mutate(Prot.found = gsub("CaO19.3767", "PEP1", Prot.found))

# issue with PEP1, ambiguous ID in FungiFun, used CaO19.3767 as identifier
f5_go2 <- process_fungifun("figures/f5_GO_clust2.csv")

f5_go3 <- process_fungifun("figures/f5_GO_clust3.csv") %>% 
  mutate(Prot.found = gsub("CaO19.4937", "CHS3", Prot.found))

# issue with CHS3, ambiguous ID in FungiFun, used CaO19.4937 as identifier
f5_go4 <- process_fungifun("figures/f5_GO_clust4.csv")

# issue with ALA1, MDH1, used CaO19.5746 and CaO19.7481 as identifiers
f5_go5 <- process_fungifun("figures/f5_GO_clust5.csv") %>% 
  mutate(Prot.found = gsub("CaO19.5746", "ALA1", Prot.found), 
         Prot.found = gsub("CaO19.7481", "MDH1", Prot.found), 
         Prot.found = gsub("CaO19.3002", "RPS1", Prot.found), 
         Prot.found = gsub("CaO19.7417", "TSA1", Prot.found))

f5_go6 <- process_fungifun("figures/f5_GO_clust6.csv")
```

```{r figure5, fig.cap=f5_cap, fig.height=6}
f5 <- png::readPNG("figures/figure5.png")

# draw figure 5
ggplot() + 
  background_image(f5) + 
  theme(plot.margin = margin(t = 0.1, l = 0.1, r = 0.1, b = 0.1, 
                             unit = "inches"))

f5_cap <- "\\textbf{Identification of commonly enriched EV proteins across 
           different \\textit{C. albicans} strains.} Heatmap of proteins 
           (n = 396) identified in all four EV samples (venn overlap in Figure 
           \\@ref(fig:figure4)A). Each column represents a 
           \\textit{C. albicans} strain and each row represents a protein. The 
           colours indicate the log\\textsubscript{2}(FC) of the protein for 
           that particular strain; red is EV enriched, blue is EV depleted, and 
           black is EV exclusive. Unsupervised clustering of rows and columns 
           was performed using Gower's formula via the daisy function from the 
           R package \\textit{cluster} [@maechler2019]. Proteins which were 
           exclusive to EVs from all four strains could not be clustered and 
           were seperated prior to clustering, in the group designated 
           \`\`cluster 6\". Functional enrichment analyses were performed on 
           the heatmap protein clusters using the online tool FungiFun2 
           [@priebe2015]. A selection of significantly enriched GO terms are 
           presented in the table inset. Lists of the proteins identified in 
           each cluster are shown and are also available in supplemental 
           File S5. \\textsuperscript{a} indicates where all proteins from 
           each protein group in the cluster were used for the GO analysis."
```

\newpage

```{r figure6Data, include=FALSE}
f6a_venn <- plot_venn(vlist = marker_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 0.9, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3), 
                      sub = "EV exclusive + sig. enriched", 
                      sub.fontfamily = "sans", 
                      sub.pos = c(0.5, 1.05), 
                      margin = 0.05)

f6a <- as_ggplot(f6a_venn) + 
  theme(plot.margin = unit(c(1.2,0,0,0), "lines"))

wcl_comp <- list("DAY Y" = yeast_res %>% 
                   filter(group == "WCL up" | group == "WCL ex") %>% 
                   pull(CGD_gene_name), 
                 "DAY B" = biofilm_res %>% 
                   filter(group == "WCL up" | group == "WCL ex") %>% 
                   pull(CGD_gene_name),
                 "ATCC90028" = a9_res %>% 
                   filter(group == "A9_W up" | group == "A9_W ex") %>% 
                   pull(CGD_gene_name), 
                 "ATCC10231" = a1_res %>% 
                   filter(group == "A1_W up" | group == "A1_W ex") %>% 
                   pull(CGD_gene_name))

f6b_venn <- plot_venn(vlist = wcl_comp, use_uniprot = FALSE, type = "plot", 
                      fontfamily = "sans", 
                      cat.fontfamily = "sans", 
                      cex = 0.9, 
                      cat.cex = 0.8, 
                      fill = st_pal[c(1,4,2,3)], 
                      alpha = c(0.3, 0.3, 0.3, 0.3), 
                      sub = "WCL exclusive + sig. enriched", 
                      sub.fontfamily = "sans", 
                      sub.pos = c(0.5, 1.05), 
                      margin = 0.05)

f6b <- as_ggplot(f6b_venn) + 
  theme(plot.margin = unit(c(1.2,0,0,0), "lines"))
```

```{r figure6, fig.cap=f6_cap, fig.width=2.5, fig.height=4}
ggarrange(f6a, f6b, 
          labels = c("A", "B"), 
          ncol = 1,
          nrow = 2)

f6_cap <- "\\textbf{Identification of common signficantly enriched or exclusive 
           \\textit{C. albicans} EV and WCL proteins.} Proteins identified as 
           significantly enriched or exclusive to EVs in Figure 
           \\@ref(fig:figure2) were compared to select those which were common 
           to all 4 datasets. 47 proteins were identified as potential positive 
           EV protein markers and 62 proteins as potential negative EV protein 
           markers. The data tables underlying Figure 
           \\@ref(fig:figure6) are provided in supplemental File S4. Details of 
           the 47 positive EV protein marker candidates and 62 negative 
           candidates can be found in Table \\@ref(tab:table1) and supplemental 
           Table S2 respectively."
```

\newpage

```{r figure7, fig.cap=f7_cap, fig.width=3.5, fig.height=2.8}
f7_cap <- "\\textbf{Twenty-two best potential \\textit{C. albicans} positive EV 
           protein markers.} These proteins are those underlined in Table 
           \\@ref(tab:table1). They are are enriched or exclusive to EVs 
           isolated from all of the \\textit{C. albicans} strains investigated 
           in this study. The proteins fit within the EV biomarker categories 
           1 and 2 from MISEV2018 which includes transmembrane, 
           GPI-anchored, and cytosolic proteins that are EV enriched 
           [@thery2018]."

f7 <- png::readPNG("figures/figure7.png")

ggplot() + 
  background_image(f7)
```

\newpage

```{r figure8, fig.cap=f8_cap, fig.width=7.2, fig.height=6}
f8_cap <- "\\textbf{Predicted topology and palmitoylation sites of 
           \\textit{C. albicans} MCC/eisosome proteins Sur7 and Evp1 
           (orf19.6741) 
           compared to the human tetraspanin CD81.} The presence of 
           transmembrane (TM) domains and palmitoylation sites shown on the 
           left side of the figure were predicted using the online tools 
           TOPCONS2 (topcons.cbr.su.se) [@tsirigos2015] and CSS-PALM 4.0 
           (csspalm.biocuckoo.org) [@ren2008] respectively. For CD81, 
           palmitoylation sites experimentally determined in previous 
           literature are shown [@delandre2009]. The right side of the figure 
           shows a cartoon representation of the TOPCONS2 consensus topology 
           prediction for the three proteins as well as the crystal structure 
           of CD81 (PDB: 5TCX) [@zimmerman2016]."

f8 <- png::readPNG("figures/figure8.png")

ggplot() + 
  background_image(f8)
```

