#' \emph{C. albicans} (ATCC10231, ATCC90028) extracellular vesicle and
#' whole cell lysate LC-MS/MS data
#'
#' @description \code{atcc} is a data.frame of a MaxQuant \code{proteinGroups.txt}
#' file generated by \href{https://doi.org}{Dawson et al. (2019)}.
#'
#' It contains label free quantification (LFQ) and intensity-based absolute
#' quantification (iBAQ) proteomics data for extracellular vesicles (EV) and whole
#' cell lysates (WCL or W) obtained from two different \emph{Candida albicans} strains;
#' \href{https://www.atcc.org/products/all/10231.aspx}{ATCC10231} (n = 3) and
#' \href{https://www.atcc.org/Products/all/90028.aspx}{ATCC90028}
#' (n = 3).
#'
#' Description about morphology/growth.
#'
#' There are 12 samples total in this data set:
#' \itemize{
#' \item 10231 EV1, EV2, EV3
#' \item 10231 W1, W2, W3
#' \item 90028 EV1, EV2, EV3
#' \item 90028 W1, W2, W3
#' }
#'
#' The EV and WCL biological replicates are paired as they are derived from
#' the same liquid culture, i.e. \code{90028_EV1} EVs were isolated from the same
#' culture from which cells were lysed to obtained \code{90028_W1} WCL.
#'
#' See \href{https://doi.org}{Dawson et al. (2019)} for more information about the
#' generation of this data set. The raw \code{proteinGroups.txt} file and the
#' script used to generate this data.frame can be found
#' \href{https://github.com/csdaw/candidaev/tree/master/data-raw/lcms}{here}.
#'
#' @details Each row represents 1 protein group. A protein group is the protein(s) which
#' can be reconstructed or identified from a set of peptides. Most protein
#' groups in this data set contain only 1 protein, but some contain 2 or more
#' ambiguous proteins e.g. CDR1 and CDR2 which can not be unambiguously
#' identified by their unique peptides.
#'
#' See the
#' \href{http://www.coxdocs.org/doku.php?id=maxquant:table:proteingrouptable}{MaxQuant documentation}
#' for a description of the columns in the \code{proteingroups.txt} output.
#'
#' A subset of relevant columns are described below:
#'
#' \describe{
#'   \item{\strong{Protein.IDs}}{UniProt accessions of protein(s) in the protein group, in descending order of peptides identified}
#'   \item{\strong{Majority.protein.IDs}}{UniProt accessions of protein(s) with at least half of peptides of first protein}
#'   \item{\strong{Peptide.counts.unique}}{Number of unique peptides associated with each protein in protein group}
#'   \item{\strong{Fasta.headers}}{Fasta header for protein(s) in protein group}
#'   \item{\strong{Number.of.proteins}}{Number of proteins in protein group}
#'   \item{\strong{Unique.peptides}}{Number of unique peptides associated with protein group. Total and individual sample numbers given}
#'   \item{\strong{Unique.sequence coverage}}{Percent of protein sequence covered by unique peptides of first protein in protein group}
#'   \item{\strong{Mol.weight.kDa}}{Molecular weight of first protein in protein group}
#'   \item{\strong{Sequence.lengths}}{Length of sequences of each protein in protein group}
#'   \item{\strong{Q.value}}{Ratio}
#'   \item{\strong{Score}}{Protein score derived from peptide posterior error probabilities (PEP)}
#'   \item{\strong{Intensity}}{Description}
#'   \item{\strong{iBAQ}}{Summed intensity}
#'   \item{\strong{LFQ.intensity}}{Description}
#'   \item{\strong{MS.MS.count}}{Description}
#'   \item{\strong{Only.identified.by.site}}{'+' = protein group was only identified by a modification site}
#'   \item{\strong{Reverse}}{'+' = protein group contains reversed peptides from decoy database}
#'   \item{\strong{Potential contamination}}{'+' = protein group is in common contaminant database}
#' }
#'
#' @format An object of class \code{data.frame} with 2451 observations (rows) and 129 variables (columns).
#'
#' @source Source goes here
"atcc"
