#' \emph{C. albicans} (DAY286 yeast) extracellular vesicle and
#' whole cell lysate LC-MS/MS data.
#'
#' @description \code{yeast} is a data.frame of a MaxQuant \code{proteinGroups.txt}
#' file generated by \href{https://doi.org}{Dawson et al. (2019)}.
#'
#' It contains label free quantification (LFQ) and intensity-based absolute
#' quantification (iBAQ) proteomics data for extracellular vesicles (EV) and whole
#' cell lysates (WCL or W) obtained from the \emph{Candida albicans} strain DAY286
#' yeast liquid cultures (n = 3 (C6-C8) and n = 1 (C3)).
#'
#' There are 8 samples total in this data set:
#'
#' \itemize{
#' \item DAY286 yeast C3.EV, C6.EV, C7.EV, C8.EV
#' \item DAY286 yeast C3.WCL, C6.WCL, C7.WCL, C8.WCL
#' }
#'
#' The EV and WCL biological replicates are paired as they are derived from
#' the same liquid culture, i.e. \code{C7} EVs were isolated from the culture medium
#' used to grow the cells which were lysed to obtain \code{C7} WCL.
#'
#' \strong{NOTE: the EVs used for the C3.EV sample were isolated using a
#' different protocol to the other samples. Hence the C3.EV and C3.WCL samples
#' should be removed prior to analysis of the other biological replicates.}
#'
#' See \href{https://doi.org}{Dawson et al. (2019)} for more information about the
#' generation of this data set. The raw \code{proteinGroups.txt} file and the
#' script used to generate this data.frame can be found
#' \href{https://github.com/csdaw/candidaev/tree/master/data-raw/lcms}{here}.
#'
#' @details Each row represents 1 protein group. A protein group is the protein(s) which
#' can be reconstructed or identified from a set of peptides. Most protein
#' groups in this data set contain only 1 protein, but some contain 2 or more
#' ambiguous proteins e.g. CDR1 and CDR2 which can not be unambiguously
#' identified by their unique peptides.
#'
#' See the
#' \href{http://www.coxdocs.org/doku.php?id=maxquant:table:proteingrouptable}{MaxQuant documentation}
#' for a description of the columns in the \code{proteingroups.txt} output.
#'
#' A subset of relevant columns are described below:
#'
#' \describe{
#'   \item{\strong{Protein.IDs}}{UniProt accessions of protein(s) in the protein group, in descending order of peptides identified}
#'   \item{\strong{Majority.protein.IDs}}{UniProt accessions of protein(s) with at least half of peptides of first protein}
#'   \item{\strong{Peptide.counts.unique}}{Number of unique peptides associated with each protein in protein group}
#'   \item{\strong{Fasta.headers}}{Fasta header for protein(s) in protein group}
#'   \item{\strong{Number.of.proteins}}{Number of proteins in protein group}
#'   \item{\strong{Unique.peptides}}{Number of unique peptides associated with protein group. Total and individual sample numbers given}
#'   \item{\strong{Unique.sequence coverage}}{Percent of protein sequence covered by unique peptides of first protein in protein group}
#'   \item{\strong{Mol.weight.kDa}}{Molecular weight of first protein in protein group}
#'   \item{\strong{Sequence.lengths}}{Length of sequences of each protein in protein group}
#'   \item{\strong{Q.value}}{Ratio of reverse to forward protein groups}
#'   \item{\strong{Score}}{Protein score derived from peptide posterior error probabilities (PEP)}
#'   \item{\strong{Intensity}}{Sum of the XIC of all isotopic clusters associated with the identified amino acid sequence}
#'   \item{\strong{iBAQ}}{Intensity divided by number of theoretical peptides produced by digestion of a protein}
#'   \item{\strong{LFQ.intensity}}{Intensity value output from the \href{https://doi.org/10.1074/mcp.M113.031591}{MaxLFQ}
#'   algorithms in MaxQuant, which is supposed to serve as a proxy for absolute protein abundance}
#'   \item{\strong{MS.MS.count}}{Number of sequencing events recorded for this protein sequence}
#'   \item{\strong{Only.identified.by.site}}{'+' indicates th protein group was only identified by a modification site}
#'   \item{\strong{Reverse}}{'+' indicates the protein group contains reversed peptides from decoy database}
#'   \item{\strong{Potential contamination}}{'+' indicates the protein group is in common contaminant database}
#' }
#'
#' @format An object of class \code{data.frame} with 1997 observations (rows) and 97 variables (columns).
#'
#' @source Dawson, C. S., Garcia-Ceron, D., Rajapaksha, H., Faou, P., Anderson,
#' M. A., Bleackley, M. R. (2020) Protein markers for \emph{Candida albicans}
#' EVs include claudin-like Sur7 family proteins.
#' \emph{Journal of Extracellular Vesicles}, x(x): 1750810. DOI:
#' \href{https://doi.org/10.1080/20013078.2020.1750810}{10.1080/20013078.2020.1750810}
"yeast"
